// @nextui-org/react/esm/image/image-browser.js
import _JSXStyle from "styled-jsx/style";
import React, { useMemo } from 'react';
import Link from '../link';
import useTheme from '../use-theme';
import withDefaults from '../utils/with-defaults';
import ImageBrowserHttpsIcon from './image-browser-https-icon';
import { getBrowserColors } from './styles';
import { __DEV__ } from '../utils/assertion';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  className: '',
  showFullLink: false,
  anchorProps: {},
  invert: false
};

const getHostFromUrl = url => {
  try {
    return new URL(url).host;
  } catch (e) {
    return url;
  }
};

const getTitle = (title, colors) => /*#__PURE__*/_jsxs("div", {
  className: _JSXStyle.dynamic([["3602594722", [colors.titleColor]]]) + " " + "title",
  children: [title, /*#__PURE__*/_jsx(_JSXStyle, {
    id: "3602594722",
    dynamic: [colors.titleColor],
    children: `.title.__jsx-style-dynamic-selector{color:${colors.titleColor};font-size:0.75rem;}`
  })]
});

const getAddressInput = (url, showFullLink, colors, anchorProps) => /*#__PURE__*/_jsxs("div", {
  className: _JSXStyle.dynamic([["2406907907", [colors.inputBgColor]]]) + " " + "address-input",
  children: [/*#__PURE__*/_jsx("span", {
    className: _JSXStyle.dynamic([["2406907907", [colors.inputBgColor]]]) + " " + "https",
    children: /*#__PURE__*/_jsx(ImageBrowserHttpsIcon, {})
  }), /*#__PURE__*/_jsx(Link, {
    href: url,
    title: url,
    target: "_blank",
    ...anchorProps,
    children: showFullLink ? url : getHostFromUrl(url)
  }), /*#__PURE__*/_jsx(_JSXStyle, {
    id: "2406907907",
    dynamic: [colors.inputBgColor],
    children: `.address-input.__jsx-style-dynamic-selector{height:1.75rem;max-width:60%;min-width:40%;background-color:${colors.inputBgColor};color:inherit;border-radius:3px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0 10px;overflow:hidden;position:relative;}.address-input.__jsx-style-dynamic-selector *{font-size:0.75rem;color:inherit;}.address-input.__jsx-style-dynamic-selector a{max-width:90%;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;display:inline-block;color:inherit;}.https.__jsx-style-dynamic-selector{width:12px;height:12px;margin-right:5px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;margin-top:-1px;color:inherit;}`
  })]
});

const ImageBrowser = /*#__PURE__*/React.forwardRef(({
  url,
  title,
  children,
  showFullLink,
  invert,
  anchorProps,
  className,
  ...props
}, ref) => {
  const theme = useTheme();
  const colors = useMemo(() => getBrowserColors(invert, theme.palette), [invert, theme.palette]);
  const input = useMemo(() => {
    if (url) return getAddressInput(url, showFullLink, colors, anchorProps);
    if (title) return getTitle(title, colors);
    return null;
  }, [url, showFullLink, title, colors, anchorProps]);
  return /*#__PURE__*/_jsxs("div", {
    ref: ref,
    ...props,
    className: _JSXStyle.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + (props && props.className != null && props.className || `browser ${className}`),
    children: [/*#__PURE__*/_jsxs("header", {
      className: _JSXStyle.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]),
      children: [/*#__PURE__*/_jsxs("div", {
        className: _JSXStyle.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "traffic",
        children: [/*#__PURE__*/_jsx("span", {
          className: _JSXStyle.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "close"
        }), /*#__PURE__*/_jsx("span", {
          className: _JSXStyle.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "mini"
        }), /*#__PURE__*/_jsx("span", {
          className: _JSXStyle.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "full"
        })]
      }), input]
    }), children, /*#__PURE__*/_jsx(_JSXStyle, {
      id: "811665925",
      dynamic: [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf],
      children: `.browser.__jsx-style-dynamic-selector{background-color:transparent;box-shadow:${theme.expressiveness.shadowLarge};width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:100%;margin:0 auto;border-radius:${theme.layout.radius};overflow:hidden;}.browser.__jsx-style-dynamic-selector .image{border-top-left-radius:0;border-top-right-radius:0;}header.__jsx-style-dynamic-selector{height:2.5rem;width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;position:relative;color:${colors.color};background-color:${colors.barBgColor};border-bottom:1px solid ${colors.borderColor};}.traffic.__jsx-style-dynamic-selector{width:auto;position:absolute;left:${theme.layout.gapHalf};top:50%;-webkit-transform:translateY(-50%);-ms-transform:translateY(-50%);transform:translateY(-50%);bottom:0;height:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.traffic.__jsx-style-dynamic-selector span.__jsx-style-dynamic-selector{border-radius:50%;width:0.75rem;height:0.75rem;display:inline-block;margin-right:0.5rem;}.close.__jsx-style-dynamic-selector{background-color:#ff5f56;}.mini.__jsx-style-dynamic-selector{background-color:#ffbd2e;}.full.__jsx-style-dynamic-selector{background-color:#27c93f;}`
    })]
  });
});

if (__DEV__) {
  ImageBrowser.displayName = 'ImageBrowser';
}

export default withDefaults(ImageBrowser, defaultProps);
