// @nextui-org/react/cjs/image/image-browser.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _link = _interopRequireDefault(require("../link"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _imageBrowserHttpsIcon = _interopRequireDefault(require("./image-browser-https-icon"));

var _styles = require("./styles");

var _assertion = require("../utils/assertion");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  className: '',
  showFullLink: false,
  anchorProps: {},
  invert: false
};

const getHostFromUrl = url => {
  try {
    return new URL(url).host;
  } catch (e) {
    return url;
  }
};

const getTitle = (title, colors) => /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
  className: _style.default.dynamic([["3602594722", [colors.titleColor]]]) + " " + "title",
  children: [title, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
    id: "3602594722",
    dynamic: [colors.titleColor],
    children: `.title.__jsx-style-dynamic-selector{color:${colors.titleColor};font-size:0.75rem;}`
  })]
});

const getAddressInput = (url, showFullLink, colors, anchorProps) => /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
  className: _style.default.dynamic([["2406907907", [colors.inputBgColor]]]) + " " + "address-input",
  children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
    className: _style.default.dynamic([["2406907907", [colors.inputBgColor]]]) + " " + "https",
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_imageBrowserHttpsIcon.default, {})
  }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_link.default, {
    href: url,
    title: url,
    target: "_blank",
    ...anchorProps,
    children: showFullLink ? url : getHostFromUrl(url)
  }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
    id: "2406907907",
    dynamic: [colors.inputBgColor],
    children: `.address-input.__jsx-style-dynamic-selector{height:1.75rem;max-width:60%;min-width:40%;background-color:${colors.inputBgColor};color:inherit;border-radius:3px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0 10px;overflow:hidden;position:relative;}.address-input.__jsx-style-dynamic-selector *{font-size:0.75rem;color:inherit;}.address-input.__jsx-style-dynamic-selector a{max-width:90%;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;display:inline-block;color:inherit;}.https.__jsx-style-dynamic-selector{width:12px;height:12px;margin-right:5px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;margin-top:-1px;color:inherit;}`
  })]
});

const ImageBrowser = /*#__PURE__*/_react.default.forwardRef(({
  url,
  title,
  children,
  showFullLink,
  invert,
  anchorProps,
  className,
  ...props
}, ref) => {
  const theme = (0, _useTheme.default)();
  const colors = (0, _react.useMemo)(() => (0, _styles.getBrowserColors)(invert, theme.palette), [invert, theme.palette]);
  const input = (0, _react.useMemo)(() => {
    if (url) return getAddressInput(url, showFullLink, colors, anchorProps);
    if (title) return getTitle(title, colors);
    return null;
  }, [url, showFullLink, title, colors, anchorProps]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    ref: ref,
    ...props,
    className: _style.default.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + (props && props.className != null && props.className || `browser ${className}`),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("header", {
      className: _style.default.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]),
      children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: _style.default.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "traffic",
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
          className: _style.default.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "close"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
          className: _style.default.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "mini"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
          className: _style.default.dynamic([["811665925", [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf]]]) + " " + "full"
        })]
      }), input]
    }), children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "811665925",
      dynamic: [theme.expressiveness.shadowLarge, theme.layout.radius, colors.color, colors.barBgColor, colors.borderColor, theme.layout.gapHalf],
      children: `.browser.__jsx-style-dynamic-selector{background-color:transparent;box-shadow:${theme.expressiveness.shadowLarge};width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:100%;margin:0 auto;border-radius:${theme.layout.radius};overflow:hidden;}.browser.__jsx-style-dynamic-selector .image{border-top-left-radius:0;border-top-right-radius:0;}header.__jsx-style-dynamic-selector{height:2.5rem;width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;position:relative;color:${colors.color};background-color:${colors.barBgColor};border-bottom:1px solid ${colors.borderColor};}.traffic.__jsx-style-dynamic-selector{width:auto;position:absolute;left:${theme.layout.gapHalf};top:50%;-webkit-transform:translateY(-50%);-ms-transform:translateY(-50%);transform:translateY(-50%);bottom:0;height:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.traffic.__jsx-style-dynamic-selector span.__jsx-style-dynamic-selector{border-radius:50%;width:0.75rem;height:0.75rem;display:inline-block;margin-right:0.5rem;}.close.__jsx-style-dynamic-selector{background-color:#ff5f56;}.mini.__jsx-style-dynamic-selector{background-color:#ffbd2e;}.full.__jsx-style-dynamic-selector{background-color:#27c93f;}`
    })]
  });
});

if (_assertion.__DEV__) {
  ImageBrowser.displayName = 'ImageBrowser';
}

var _default = (0, _withDefaults.default)(ImageBrowser, defaultProps);

exports.default = _default;
module.exports = exports.default;
