// @nextui-org/react/cjs/checkbox/checkbox-group.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _checkboxContext = require("./checkbox-context");

var _useWarning = _interopRequireDefault(require("../use-warning"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _assertion = require("../utils/assertion");

var _styles = require("./styles");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  color: 'primary',
  textColor: 'default',
  disabled: false,
  size: 'medium',
  className: '',
  row: false
};

const CheckboxGroup = ({
  color,
  textColor,
  disabled,
  onChange,
  value,
  size,
  row,
  children,
  className,
  ...props
}) => {
  const [selfVal, setSelfVal] = (0, _react.useState)([]);

  if (!value && _assertion.__DEV__) {
    value = [];
    (0, _useWarning.default)('Props "value" is required.', 'Checkbox Group');
  }

  const updateState = (val, checked) => {
    const removed = selfVal.filter(v => v !== val);
    const next = checked ? [...removed, val] : removed;
    setSelfVal(next);
    onChange && onChange(next);
  };

  const providerValue = (0, _react.useMemo)(() => {
    return {
      updateState,
      color,
      textColor,
      disabledAll: disabled,
      inGroup: true,
      values: selfVal
    };
  }, [disabled, selfVal]);
  const fontSize = (0, _react.useMemo)(() => (0, _styles.getCheckboxSize)(size), [size]);
  const groupGap = `calc(${fontSize} * 1)`;
  (0, _react.useEffect)(() => {
    setSelfVal(value);
  }, [value.join(',')]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_checkboxContext.CheckboxContext.Provider, {
    value: providerValue,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      role: "group",
      ...props,
      className: _style.default.dynamic([["801134383", [row ? 0 : groupGap, row ? groupGap : 0, fontSize, row ? 'row' : 'column']]]) + " " + (props && props.className != null && props.className || `checkbox-group ${className}`),
      children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
        id: "801134383",
        dynamic: [row ? 0 : groupGap, row ? groupGap : 0, fontSize, row ? 'row' : 'column'],
        children: `.checkbox-group.__jsx-style-dynamic-selector .checkbox{margin-top:${row ? 0 : groupGap};margin-right:${row ? groupGap : 0};--checkbox-size:${fontSize};}.checkbox-group.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:${row ? 'row' : 'column'};-ms-flex-direction:${row ? 'row' : 'column'};flex-direction:${row ? 'row' : 'column'};}`
      })]
    })
  });
};

var _default = (0, _withDefaults.default)(CheckboxGroup, defaultProps);

exports.default = _default;
module.exports = exports.default;
