// @nextui-org/react/cjs/radio/radio.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _radioContext = require("./radio-context");

var _radioGroup = require("./radio-group");

var _radioDescription = _interopRequireDefault(require("./radio-description"));

var _collections = require("../utils/collections");

var _useWarning = _interopRequireDefault(require("../use-warning"));

var _useKeyboard = _interopRequireWildcard(require("../use-keyboard"));

var _styles = require("../utils/styles");

var _color = require("../utils/color");

var _assertion = require("../utils/assertion");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  size: 'medium',
  color: 'primary',
  textColor: 'default',
  disabled: false,
  squared: false,
  className: ''
};

const Radio = ({
  className,
  checked,
  onChange,
  squared,
  disabled,
  size,
  color,
  textColor,
  value: radioValue,
  children,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  const [selfChecked, setSelfChecked] = (0, _react.useState)(!!checked);
  const {
    value: groupValue,
    disabledAll,
    inGroup,
    color: groupColor,
    textColor: textGroupColor,
    updateState
  } = (0, _radioContext.useRadioContext)();
  const [withoutDescChildren, DescChildren] = (0, _collections.pickChild)(children, _radioDescription.default);

  if (inGroup && _assertion.__DEV__) {
    if (checked !== undefined) {
      (0, _useWarning.default)('Remove props "checked" if in the Radio.Group.', 'Radio');
    }

    if (radioValue === undefined) {
      (0, _useWarning.default)('Props "value" must be deinfed if in the Radio.Group.', 'Radio');
    }
  }

  (0, _react.useEffect)(() => {
    setSelfChecked(groupValue === radioValue);
  }, [groupValue, radioValue]);
  const fontSize = (0, _react.useMemo)(() => (0, _radioGroup.getRadioSize)(size), [size]);
  const isDisabled = (0, _react.useMemo)(() => disabled || disabledAll, [disabled, disabledAll]);
  const radius = squared ? '2px' : '50%';
  const radioColor = (0, _react.useMemo)(() => isDisabled ? theme.palette.accents_4 : (0, _color.getNormalColor)(color || groupColor, theme.palette, theme.palette.foreground), [color, groupColor, isDisabled, theme.palette]);
  const labelColor = (0, _react.useMemo)(() => isDisabled ? theme.palette.accents_4 : (0, _color.getNormalColor)(textColor || textGroupColor, theme.palette, theme.palette.text), [textColor, textGroupColor, isDisabled, theme.palette]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = (0, _styles.getFocusStyles)(theme);

  const changeHandler = event => {
    if (isDisabled) return;
    const selfEvent = {
      target: {
        checked: !selfChecked
      },
      stopPropagation: event.stopPropagation,
      preventDefault: event.preventDefault,
      nativeEvent: event
    };
    setSelfChecked(!selfChecked);

    if (inGroup) {
      updateState && updateState(radioValue);
    }

    onChange && onChange(selfEvent);
  };

  const {
    bindings
  } = (0, _useKeyboard.default)(event => {
    changeHandler(event);
  }, [_useKeyboard.KeyCode.Enter, _useKeyboard.KeyCode.Space], {
    disableGlobalEvent: true
  });
  (0, _react.useEffect)(() => {
    if (checked === undefined) return;
    setSelfChecked(Boolean(checked));
  }, [checked]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("label", {
    role: "radio",
    "aria-checked": selfChecked,
    ...props,
    ...bindings,
    className: _style.default.dynamic([["3122260096", [fontSize, labelColor, isDisabled ? 'not-allowed' : 'pointer', radius, radius, theme.palette.border, radioColor, theme.palette.border]]]) + " " + (bindings && bindings.className != null && bindings.className || props && props.className != null && props.className || `radio ${className}`),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("input", {
      type: "radio",
      tabIndex: -1,
      value: radioValue,
      checked: selfChecked,
      onChange: changeHandler,
      ...props,
      className: _style.default.dynamic([["3122260096", [fontSize, labelColor, isDisabled ? 'not-allowed' : 'pointer', radius, radius, theme.palette.border, radioColor, theme.palette.border]]]) + " " + (props && props.className != null && props.className || "")
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("span", {
      className: _style.default.dynamic([["3122260096", [fontSize, labelColor, isDisabled ? 'not-allowed' : 'pointer', radius, radius, theme.palette.border, radioColor, theme.palette.border]]]) + " " + "name",
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
        tabIndex: disabled ? -1 : 0,
        className: _style.default.dynamic([["3122260096", [fontSize, labelColor, isDisabled ? 'not-allowed' : 'pointer', radius, radius, theme.palette.border, radioColor, theme.palette.border]]]) + " " + `point ${selfChecked ? 'active' : ''} ${isDisabled ? 'disabled' : ''} ${focusClassName}`
      }), withoutDescChildren]
    }), DescChildren && DescChildren, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3122260096",
      dynamic: [fontSize, labelColor, isDisabled ? 'not-allowed' : 'pointer', radius, radius, theme.palette.border, radioColor, theme.palette.border],
      children: `input.__jsx-style-dynamic-selector{opacity:0;overflow:hidden;width:1px;height:1px;top:-1000px;right:-1000px;position:fixed;}.radio.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:initial;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;position:relative;--radio-size:${fontSize};}label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;color:${labelColor};cursor:${isDisabled ? 'not-allowed' : 'pointer'};}.name.__jsx-style-dynamic-selector{font-size:var(--radio-size);-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.point.__jsx-style-dynamic-selector{width:var(--radio-size);height:var(--radio-size);border-radius:${radius};-webkit-transition:all 0.25s ease;transition:all 0.25s ease;position:relative;display:inline-block;margin-right:calc(var(--radio-size) * 0.375);}.point.__jsx-style-dynamic-selector:after{content:'';display:block;position:absolute;width:var(--radio-size);height:var(--radio-size);border-radius:${radius};box-sizing:border-box;-webkit-transition:all 0.25s ease;transition:all 0.25s ease;border:2px solid ${theme.palette.border};}.point.active.__jsx-style-dynamic-selector:after{border:calc(var(--radio-size) * 0.34) solid ${radioColor};}label.__jsx-style-dynamic-selector:hover .point.__jsx-style-dynamic-selector:not(.active):not(.disabled){background:${theme.palette.border};}`
    }), focusStyles]
  });
};

Radio.defaultProps = defaultProps;

if (_assertion.__DEV__) {
  Radio.displayName = 'NextUI - Radio';
}

var _default = Radio;
exports.default = _default;
module.exports = exports.default;
