// @nextui-org/react/cjs/radio/radio-group.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = exports.getRadioSize = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _radioContext = require("./radio-context");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  disabled: false,
  size: 'medium',
  color: 'primary',
  textColor: 'default',
  className: '',
  row: false
};

const getRadioSize = size => {
  const sizes = {
    mini: '14px',
    small: '16px',
    medium: '20px',
    large: '24px',
    xlarge: '28px'
  };
  if (typeof size === 'number') return `${size}px`;
  return sizes[size];
};

exports.getRadioSize = getRadioSize;

const RadioGroup = ({
  disabled,
  onChange,
  value,
  size,
  color,
  textColor,
  children,
  className,
  initialValue,
  row,
  ...props
}) => {
  const [selfVal, setSelfVal] = (0, _react.useState)(initialValue);

  const updateState = nextValue => {
    setSelfVal(nextValue);
    onChange && onChange(nextValue);
  };

  const providerValue = (0, _react.useMemo)(() => {
    return {
      updateState,
      disabledAll: disabled,
      inGroup: true,
      color,
      textColor,
      value: selfVal
    };
  }, [disabled, selfVal]);
  const fontSize = (0, _react.useMemo)(() => getRadioSize(size), [size]);
  const groupGap = `calc(${fontSize} * 1)`;
  (0, _react.useEffect)(() => {
    if (value === undefined) return;
    setSelfVal(value);
  }, [value]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_radioContext.RadioContext.Provider, {
    value: providerValue,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("fieldset", {
      role: "radiogroup",
      ...props,
      className: _style.default.dynamic([["3133287453", [row ? 'row' : 'column', row ? 0 : groupGap, row ? groupGap : 0, fontSize]]]) + " " + (props && props.className != null && props.className || `radio-group ${className}`),
      children: children
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3133287453",
      dynamic: [row ? 'row' : 'column', row ? 0 : groupGap, row ? groupGap : 0, fontSize],
      children: `fieldset.__jsx-style-dynamic-selector{border:0;margin:0;padding:0;}.radio-group.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:${row ? 'row' : 'column'};-ms-flex-direction:${row ? 'row' : 'column'};flex-direction:${row ? 'row' : 'column'};}.radio-group.__jsx-style-dynamic-selector .radio{margin-top:${row ? 0 : groupGap};margin-right:${row ? groupGap : 0};--radio-size:${fontSize};}`
    })]
  });
};

var _default = (0, _withDefaults.default)(RadioGroup, defaultProps);

exports.default = _default;
