// @nextui-org/react/cjs/user/user-link.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireDefault(require("react"));

var _link = _interopRequireDefault(require("../link"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _assertion = require("../utils/assertion");

var _jsxRuntime = require("react/jsx-runtime");

const defaultProps = {
  className: ''
};

const UserLink = /*#__PURE__*/_react.default.forwardRef(({
  href,
  className,
  children,
  ...props
}, ref) => {
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", { ...props,
    className: "jsx-3263947648" + " " + (props && props.className != null && props.className || className || ""),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_link.default, {
      ref: ref,
      href: href,
      color: true,
      target: "_blank",
      rel: "noopener",
      children: children
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3263947648",
      children: "div.jsx-3263947648 a:hover{opacity:0.7;}"
    })]
  });
});

if (_assertion.__DEV__) {
  UserLink.displayName = 'UserLink';
}

const MemoUserLink = /*#__PURE__*/_react.default.memo(UserLink);

var _default = (0, _withDefaults.default)(MemoUserLink, defaultProps);

exports.default = _default;
module.exports = exports.default;
