// @nextui-org/react/cjs/user/user.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _jsxRuntime = require("react/jsx-runtime");

const defaultProps = {
  className: '',
  size: 32,
  squared: false,
  bordered: false
};

const User = ({
  src,
  text,
  name,
  children,
  className,
  altText,
  color,
  squared,
  bordered,
  size,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", { ...props,
    className: _style.default.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + (props && props.className != null && props.className || `user ${className}`),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_index.Avatar, {
      src: src,
      color: color,
      squared: squared,
      bordered: bordered,
      text: text,
      size: size,
      alt: altText
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: _style.default.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + "names",
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
        className: _style.default.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + "name",
        children: name
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
        className: _style.default.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + "social",
        children: children
      })]
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "1525555450",
      dynamic: [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4],
      children: `.user.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;padding:0 ${theme.layout.gapHalf};-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:100%;}.names.__jsx-style-dynamic-selector{margin-left:${theme.layout.gapHalf};display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;white-space:nowrap;}.name.__jsx-style-dynamic-selector{font-size:0.89rem;color:${theme.palette.text};line-height:1.1rem;text-transform:capitalize;font-weight:500;max-width:15rem;text-overflow:ellipsis;overflow:hidden;}.social.__jsx-style-dynamic-selector{font-size:0.75rem;color:${theme.palette.accents_4};}.social.__jsx-style-dynamic-selector *:first-child{margin-top:0;}.social.__jsx-style-dynamic-selector *:last-child{margin-bottom:0;}`
    })]
  });
};

User.defaultProps = defaultProps;

var _default = /*#__PURE__*/_react.default.memo(User);

exports.default = _default;
module.exports = exports.default;
