// @nextui-org/react/cjs/theme/dark.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = exports.themes = exports.layout = exports.breakpoints = exports.font = exports.expressiveness = exports.palette = void 0;

var _shared = require("./shared");

var _common = _interopRequireDefault(require("./common"));

const palette = { ..._common.default,
  accents_1: '#111',
  accents_2: '#333',
  accents_3: '#444',
  accents_4: '#666',
  accents_5: '#888',
  accents_6: '#999',
  accents_7: '#eaeaea',
  accents_8: '#fafafa',
  text: '#fff',
  background: '#000',
  foreground: '#fff',
  code: '#77C8E6',
  border: '#333',
  selection: '#ff4ecd'
};
exports.palette = palette;
const expressiveness = {
  linkStyle: 'none',
  linkHoverStyle: 'none',
  dropdownBoxShadow: '0 0 0 1px #333',
  scrollerStart: 'rgba(255, 255, 255, 1)',
  scrollerEnd: 'rgba(255, 255, 255, 0)',
  shadowSmall: '0 10px 20px -10px rgba(0, 0, 0, 0.1)',
  shadowMedium: '0 15px 22px -10px rgba(0, 0, 0, 0.1)',
  shadowLarge: '0 20px 24px -10px rgba(0, 0, 0, 0.1)',
  portalOpacity: 0.75
};
exports.expressiveness = expressiveness;
const font = _shared.defaultFont;
exports.font = font;
const breakpoints = _shared.defaultBreakpoints;
exports.breakpoints = breakpoints;
const layout = _shared.defaultLayout;
exports.layout = layout;
const themes = {
  type: 'dark',
  font,
  layout,
  palette,
  breakpoints,
  expressiveness
};
exports.themes = themes;
var _default = themes;
exports.default = _default;
