// @nextui-org/react/cjs/theme/default.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = exports.themes = exports.layout = exports.breakpoints = exports.font = exports.expressiveness = exports.palette = void 0;

var _shared = require("./shared");

var _common = _interopRequireDefault(require("./common"));

const palette = { ..._common.default,
  accents_1: '#f6f6f6',
  accents_2: '#eaeaea',
  accents_3: '#999',
  accents_4: '#888',
  accents_5: '#666',
  accents_6: '#444',
  accents_7: '#333',
  accents_8: '#111',
  text: '#333',
  white: '#fff',
  background: '#fff',
  foreground: '#000',
  code: '#f81ce5',
  border: '#eaeaea',
  selection: '#0070f3'
};
exports.palette = palette;
const expressiveness = {
  linkStyle: 'none',
  linkHoverStyle: 'none',
  dropdownBoxShadow: '0 4px 4px 0 rgba(0, 0, 0, 0.02)',
  scrollerStart: 'rgba(255, 255, 255, 1)',
  scrollerEnd: 'rgba(255, 255, 255, 0)',
  shadowSmall: '0 5px 20px -5px rgba(0, 0, 0, .1)',
  shadowMedium: '0 8px 30px rgba(0, 0, 0, 0.15)',
  shadowLarge: '0 30px 60px rgba(0, 0, 0, 0.15)',
  portalOpacity: 0.25
};
exports.expressiveness = expressiveness;
const font = _shared.defaultFont;
exports.font = font;
const breakpoints = _shared.defaultBreakpoints;
exports.breakpoints = breakpoints;
const layout = _shared.defaultLayout;
exports.layout = layout;
const themes = {
  type: 'light',
  font,
  layout,
  palette,
  breakpoints,
  expressiveness
};
exports.themes = themes;
var _default = themes;
exports.default = _default;
