// @nextui-org/react/cjs/theme/shared.js
"use strict";

exports.__esModule = true;
exports.defaultLayout = exports.defaultBreakpoints = exports.defaultFont = void 0;
const defaultFont = {
  sans: '-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
  mono: 'Menlo, Monaco, Lucida Console, Liberation Mono, DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace'
};
exports.defaultFont = defaultFont;
const defaultBreakpoints = {
  xs: {
    min: '0',
    max: '650px'
  },
  sm: {
    min: '650px',
    max: '960px'
  },
  md: {
    min: '960px',
    max: '1280px'
  },
  lg: {
    min: '1280px',
    max: '1400px'
  },
  xl: {
    min: '1400px',
    max: '1920px'
  }
};
exports.defaultBreakpoints = defaultBreakpoints;
const defaultLayout = {
  gap: '16pt',
  gapNegative: '-16pt',
  gapHalf: '8pt',
  gapHalfNegative: '-8pt',
  gapQuarter: '4pt',
  gapQuarterNegative: '-4pt',
  pageMargin: '16pt',
  pageWidth: '750pt',
  pageWidthWithMargin: '782pt',
  breakpointMobile: defaultBreakpoints.xs.max,
  breakpointTablet: defaultBreakpoints.sm.max,
  radius: '14px',
  stroke: '2px'
};
exports.defaultLayout = defaultLayout;
