// @nextui-org/react/cjs/theme/utils.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.switchTheme = exports.getThemeByType = exports.mergeTheme = void 0;

var _useWarning = _interopRequireDefault(require("../use-warning"));

var _assertion = require("../utils/assertion");

var _object = require("../utils/object");

var _dark = _interopRequireDefault(require("./dark"));

var _default = _interopRequireDefault(require("./default"));

const mergeTheme = (current, custom) => {
  if (!custom) return current;

  if (typeof custom === 'function') {
    const merged = custom(current);

    if ((!merged || typeof merged !== 'object') && _assertion.__DEV__) {
      (0, _useWarning.default)('The theme function must return object value.');
    }

    return merged;
  }

  return (0, _object.deepMergeObject)(current, custom);
};

exports.mergeTheme = mergeTheme;

const getThemeByType = type => {
  const themes = {
    light: _default.default,
    dark: _dark.default
  };
  return themes[type];
};

exports.getThemeByType = getThemeByType;

const switchTheme = mergedTheme => {
  return { ...getThemeByType(mergedTheme.type),
    ...mergedTheme
  };
};

exports.switchTheme = switchTheme;
