// @nextui-org/react/esm/spacer/spacer.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import withDefaults from '../utils/with-defaults';
import { getMargin } from '../utils/dimensions';
import { jsx as _jsx } from "react/jsx-runtime";
const defaultProps = {
  x: 1,
  y: 1,
  inline: false,
  className: ''
};

const Spacer = ({
  x,
  y,
  inline,
  className,
  ...props
}) => {
  const left = getMargin(x);
  const top = getMargin(y);
  return /*#__PURE__*/_jsx("span", { ...props,
    className: _JSXStyle.dynamic([["782780554", [inline ? 'inline-block' : 'block', left, top]]]) + " " + (props && props.className != null && props.className || className || ""),
    children: /*#__PURE__*/_jsx(_JSXStyle, {
      id: "782780554",
      dynamic: [inline ? 'inline-block' : 'block', left, top],
      children: `span.__jsx-style-dynamic-selector{display:${inline ? 'inline-block' : 'block'};height:1px;width:1px;margin-left:${left};margin-top:${top};}`
    })
  });
};

const MemoSpacer = /*#__PURE__*/React.memo(Spacer);
export default withDefaults(MemoSpacer, defaultProps);
