// @nextui-org/react/cjs/card/card-body.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _cardContext = require("./card-context");

var _clsx = _interopRequireDefault(require("../utils/clsx"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  width: '100%',
  height: 'auto',
  justify: 'inherit',
  alignItems: 'inherit',
  alignContent: 'inherit',
  direction: 'column',
  noPadding: false,
  className: ''
};

const CardBody = ({
  className,
  justify,
  alignContent,
  alignItems,
  noPadding: noPaddingProp,
  direction,
  width,
  height,
  children,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  const {
    noPadding: noPaddingContext
  } = (0, _react.useContext)(_cardContext.CardContext);
  const noPadding = (0, _react.useMemo)(() => {
    return noPaddingContext !== undefined ? noPaddingContext : noPaddingProp;
  }, [noPaddingProp, noPaddingContext]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", { ...props,
    className: _style.default.dynamic([["3689227210", [width, height, direction, justify, alignItems, alignContent, theme.layout.gapHalf, theme.layout.gap]]]) + " " + (props && props.className != null && props.className || (0, _clsx.default)('card-body', {
      'no-padding': noPadding
    }, className) || ""),
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3689227210",
      dynamic: [width, height, direction, justify, alignItems, alignContent, theme.layout.gapHalf, theme.layout.gap],
      children: `.card-body.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;width:${width};height:${height};-webkit-flex-direction:${direction};-ms-flex-direction:${direction};flex-direction:${direction};-webkit-box-pack:${justify};-webkit-justify-content:${justify};-ms-flex-pack:${justify};justify-content:${justify};-webkit-align-items:${alignItems};-webkit-box-align:${alignItems};-ms-flex-align:${alignItems};align-items:${alignItems};-webkit-align-content:${alignContent};-ms-flex-line-pack:${alignContent};align-content:${alignContent};padding:${theme.layout.gapHalf} calc(${theme.layout.gap} + 0.25rem);overflow-y:auto;position:relative;text-align:left;}.card-body.no-padding.__jsx-style-dynamic-selector{padding:0;}.card-body.__jsx-style-dynamic-selector>*:first-child{margin-top:0;}.card-body.__jsx-style-dynamic-selector>*:last-child{margin-bottom:0;}`
    })]
  });
};

const MemoCardBody = /*#__PURE__*/_react.default.memo(CardBody);

var _default = (0, _withDefaults.default)(MemoCardBody, defaultProps);

exports.default = _default;
module.exports = exports.default;
