// @nextui-org/react/cjs/card/card-footer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _clsx = _interopRequireDefault(require("../utils/clsx"));

var _cardContext = require("./card-context");

var _color = require("../utils/color");

var _dimensions = require("../utils/dimensions");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  autoMargin: false,
  blur: false,
  border: false,
  width: '100%',
  height: 'auto',
  noPadding: false,
  borderWeight: 'light',
  className: ''
};

const CardFooter = ({
  children,
  blur,
  className,
  color,
  width,
  height,
  border: borderProp,
  borderColor,
  borderWeight,
  noPadding,
  autoMargin: autoMarginProp,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  const {
    background,
    autoMargin: autoMarginContext
  } = (0, _react.useContext)(_cardContext.CardContext);
  const autoMargin = (0, _react.useMemo)(() => {
    return autoMarginContext !== undefined ? autoMarginContext : autoMarginProp;
  }, [autoMarginProp, autoMarginContext]);
  const bgColor = (0, _react.useMemo)(() => {
    if (color) {
      return (0, _color.getNormalColor)(color, theme.palette);
    }

    return background || theme.palette.background;
  }, [color, theme.palette, background]);
  const border = (0, _react.useMemo)(() => {
    if (!borderProp) return 'none';
    return `${(0, _dimensions.getNormalWeight)(borderWeight)} solid ${(0, _color.getNormalColor)(borderColor, theme.palette, theme.palette.border)}`;
  }, [borderWeight, theme.palette, borderColor, borderProp]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", { ...props,
    className: _style.default.dynamic([["135585066", [width, height, theme.layout.gapHalf, theme.layout.gap, bgColor, border, theme.layout.radius, theme.layout.radius, (0, _color.addColorAlpha)(bgColor, 0.4), theme.layout.gapQuarter]]]) + " " + (props && props.className != null && props.className || (0, _clsx.default)('card-footer', {
      'auto-margin': autoMargin,
      blur,
      'no-padding': noPadding
    }, className) || ""),
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "135585066",
      dynamic: [width, height, theme.layout.gapHalf, theme.layout.gap, bgColor, border, theme.layout.radius, theme.layout.radius, (0, _color.addColorAlpha)(bgColor, 0.4), theme.layout.gapQuarter],
      children: `.card-footer.__jsx-style-dynamic-selector{width:${width};height:${height};padding:${theme.layout.gapHalf} ${theme.layout.gap};display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;overflow:hidden;color:inherit;background-color:${bgColor};font-size:0.875rem;border-top:${border};border-bottom-left-radius:${theme.layout.radius};border-bottom-right-radius:${theme.layout.radius};}.card-footer.blur.__jsx-style-dynamic-selector{-webkit-backdrop-filter:saturate(180%) blur(10px);backdrop-filter:saturate(180%) blur(10px);background:${(0, _color.addColorAlpha)(bgColor, 0.4)};}.card-footer.no-padding.__jsx-style-dynamic-selector{padding:0;}.auto-margin.__jsx-style-dynamic-selector *{margin-top:0;margin-bottom:0;margin-right:${theme.layout.gapQuarter};}`
    })]
  });
};

const MemoCardFooter = /*#__PURE__*/_react.default.memo(CardFooter);

var _default = (0, _withDefaults.default)(MemoCardFooter, defaultProps);

exports.default = _default;
module.exports = exports.default;
