// @nextui-org/react/cjs/card/styles.js
"use strict";

exports.__esModule = true;
exports.getStyles = exports.getDripColor = void 0;

var _color = require("./../utils/color");

const getDripColor = (palette, color, bordered) => {
  const colors = {
    default: palette.accents_2,
    primary: palette.primary,
    secondary: palette.secondary,
    success: palette.success,
    warning: palette.warning,
    error: palette.error,
    gradient: (0, _color.hexFromString)(palette.gradient, palette.primary, true)
  };
  const baseColor = color === 'default' ? palette.primary : colors[color || 'default'] || (0, _color.getNormalColor)(color, palette);
  const selectedColor = bordered ? baseColor : palette.accents_6;
  if (selectedColor) return (0, _color.addColorAlpha)(selectedColor, 0.25);
  return palette.accents_2;
};

exports.getDripColor = getDripColor;

const getStyles = (color, textColor, shadow, bordered, theme) => {
  const palette = theme.palette;
  const isDark = theme.type === 'dark';
  const normalColor = (0, _color.getNormalColor)(color, palette, isDark ? palette.accents_1 : palette.background);
  const normalTextColor = (0, _color.getNormalColor)(textColor, palette, palette.text);
  const dripColor = getDripColor(palette, color, bordered);
  return {
    bgColor: color === 'default' && !shadow && !bordered ? palette.accents_1 : normalColor,
    color: normalTextColor,
    borderColor: color === 'default' ? palette.border : normalColor,
    dripColor
  };
};

exports.getStyles = getStyles;
