// @nextui-org/react/cjs/snippet/styles.js
"use strict";

exports.__esModule = true;
exports.getStyles = void 0;

var _color = require("../utils/color");

const getStyles = (type, palette, fill) => {
  const styles = {
    default: {
      color: palette.foreground,
      border: palette.border,
      bgColor: palette.background
    },
    primary: {
      color: palette.foreground,
      border: palette.border,
      bgColor: palette.primary
    },
    success: {
      color: palette.success,
      border: palette.success,
      bgColor: palette.background
    },
    warning: {
      color: palette.warning,
      border: palette.warning,
      bgColor: palette.background
    },
    error: {
      color: palette.error,
      border: palette.error,
      bgColor: palette.background
    },
    secondary: {
      color: palette.secondary,
      border: palette.secondary,
      bgColor: palette.background
    },
    invert: {
      color: palette.background,
      border: palette.foreground,
      bgColor: palette.foreground
    },
    gradient: {
      color: palette.foreground,
      border: (0, _color.hexFromString)(palette.gradient, palette.primary, true),
      bgColor: palette.gradient
    }
  };
  const filledTypes = ['success', 'warning', 'error', 'secondary'];
  const style = styles[type];
  const shouldFilled = filledTypes.includes(type);
  if (!fill || !shouldFilled) return style;
  return { ...style,
    color: style.bgColor,
    bgColor: style.color
  };
};

exports.getStyles = getStyles;
