// @nextui-org/react/cjs/spacer/spacer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireDefault(require("react"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _dimensions = require("../utils/dimensions");

var _jsxRuntime = require("react/jsx-runtime");

const defaultProps = {
  x: 1,
  y: 1,
  inline: false,
  className: ''
};

const Spacer = ({
  x,
  y,
  inline,
  className,
  ...props
}) => {
  const left = (0, _dimensions.getMargin)(x);
  const top = (0, _dimensions.getMargin)(y);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("span", { ...props,
    className: _style.default.dynamic([["782780554", [inline ? 'inline-block' : 'block', left, top]]]) + " " + (props && props.className != null && props.className || className || ""),
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "782780554",
      dynamic: [inline ? 'inline-block' : 'block', left, top],
      children: `span.__jsx-style-dynamic-selector{display:${inline ? 'inline-block' : 'block'};height:1px;width:1px;margin-left:${left};margin-top:${top};}`
    })
  });
};

const MemoSpacer = /*#__PURE__*/_react.default.memo(Spacer);

var _default = (0, _withDefaults.default)(MemoSpacer, defaultProps);

exports.default = _default;
module.exports = exports.default;
