// @nextui-org/react/esm/pagination/pagination.js
import _JSXStyle from "styled-jsx/style";
import React, { useMemo, useCallback } from 'react';
import { getPaginationSizes } from './styles';
import usePagination, { DOTS } from '../use-pagination';
import PaginationItem from './pagination-item';
import PaginationEllipsis from './pagination-ellipsis';
import PaginationIcon from './pagination-icon';
import PaginationHighlight from './pagination-highlight';
import useTheme from '../use-theme';
import clsx from '../utils/clsx';
import { getNormalColor } from '../utils/color';
import { __DEV__ } from '../utils/assertion';
import { getNormalWeight } from '../utils/dimensions';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  color: 'primary',
  size: 'medium',
  borderWeight: 'normal',
  shadow: false,
  rounded: false,
  controls: true,
  bordered: false,
  onlyDots: false,
  noMargin: false,
  initialPage: 1,
  siblings: 1,
  boundaries: 1,
  dotsJump: 5,
  total: 1,
  loop: false,
  animated: true
};

const Pagination = ({
  page,
  initialPage,
  total,
  size,
  color,
  loop,
  siblings,
  boundaries,
  shadow,
  rounded,
  animated,
  bordered,
  borderWeight,
  onlyDots,
  dotsJump,
  controls,
  noMargin,
  onChange,
  ...props
}) => {
  const {
    range,
    active,
    setPage,
    previous,
    next,
    first,
    last
  } = usePagination({
    page,
    initialPage,
    siblings: onlyDots ? 10 : siblings,
    boundaries: onlyDots ? 10 : boundaries,
    total,
    onChange
  });
  const theme = useTheme();
  const bgColor = useMemo(() => getNormalColor(color, theme.palette, theme.palette.primary), [color, theme.palette]);
  const {
    font,
    width
  } = useMemo(() => getPaginationSizes(size), [size]);
  const radius = useMemo(() => noMargin ? '0px' : rounded || onlyDots ? '100%' : '33%', [rounded, onlyDots, noMargin]);
  const weight = useMemo(() => bordered ? getNormalWeight(borderWeight) : '0px', [bordered, borderWeight]);
  const renderItem = useCallback((value, index) => {
    if (value === DOTS) {
      const isBefore = index < range.indexOf(active);
      return /*#__PURE__*/_jsx(PaginationEllipsis, {
        value: value,
        bordered: bordered,
        animated: animated,
        isBefore: isBefore,
        onlyDots: onlyDots,
        onClick: () => isBefore ? setPage(active - dotsJump >= 1 ? active - dotsJump : 1) : setPage(active + dotsJump <= total ? active + dotsJump : total)
      }, `pagination-item-${value}-${index}`);
    }

    return /*#__PURE__*/_jsx(PaginationItem, {
      value: value,
      animated: animated,
      bordered: bordered,
      active: value === active,
      onClick: () => setPage(value),
      onlyDots: onlyDots,
      children: value
    }, `pagination-item-${value}-${index}`);
  }, [total, onlyDots, active, bordered, animated]);

  const handleNext = () => {
    if (loop && active === total) {
      return first();
    }

    return next();
  };

  const handlePrevious = () => {
    if (loop && active === 1) {
      return last();
    }

    return previous();
  };

  return /*#__PURE__*/_jsxs("nav", { ...props,
    className: _JSXStyle.dynamic([["2109028265", [font, radius, weight, noMargin ? '0' : '2px', bgColor, onlyDots ? `calc(${width} / 2)` : width, font, onlyDots ? 0.8 : 0.9]]]) + " " + (props && props.className != null && props.className || clsx('pagination', {
      'no-margin': noMargin,
      bordered
    }) || ""),
    children: [controls && /*#__PURE__*/_jsx(PaginationIcon, {
      isPrev: true,
      bordered: bordered,
      animated: animated,
      onlyDots: onlyDots,
      onClick: handlePrevious,
      disabled: !loop && active === 1
    }), /*#__PURE__*/_jsx(PaginationHighlight, {
      noMargin: noMargin,
      animated: animated,
      color: bgColor,
      shadow: shadow,
      active: controls ? range.indexOf(active) + 1 : range.indexOf(active)
    }), range.map(renderItem), controls && /*#__PURE__*/_jsx(PaginationIcon, {
      bordered: bordered,
      animated: animated,
      onlyDots: onlyDots,
      onClick: handleNext,
      disabled: !loop && active === total
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "2109028265",
      dynamic: [font, radius, weight, noMargin ? '0' : '2px', bgColor, onlyDots ? `calc(${width} / 2)` : width, font, onlyDots ? 0.8 : 0.9],
      children: `.pagination.__jsx-style-dynamic-selector{margin:0;padding:0;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;position:relative;font-variant:tabular-nums;font-feature-settings:'tnum';font-size:${font};--next-ui-pagination-item-radius:${radius};--next-ui-pagination-item-border-weight:${weight};--next-ui-pagination-item-margin:${noMargin ? '0' : '2px'};--next-ui-pagination-item-color:${bgColor};--next-ui-pagination-size:${onlyDots ? `calc(${width} / 2)` : width};--next-ui-pagination-font-size:${font};--next-ui-pagination-scale-transform:${onlyDots ? 0.8 : 0.9};}.pagination.__jsx-style-dynamic-selector button:last-of-type{margin-right:0;}.no-margin.__jsx-style-dynamic-selector button:first-of-type{border-top-left-radius:33%;border-bottom-left-radius:33%;}.no-margin.__jsx-style-dynamic-selector button:last-of-type{border-top-right-radius:33%;border-bottom-right-radius:33%;}.no-margin.bordered.__jsx-style-dynamic-selector button:not(:last-child){border-right:0;}`
    })]
  });
};

Pagination.defaultProps = defaultProps;

if (__DEV__) {
  Pagination.displayName = 'NextUI - Pagination';
}

export default /*#__PURE__*/React.memo(Pagination);
