// @nextui-org/react/esm/pagination/pagination-ellipsis.js
import _JSXStyle from "styled-jsx/style";
import React, { useState } from 'react';
import PaginationItem from './pagination-item';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";

const PaginationEllipsis = ({
  value,
  isBefore,
  onlyDots,
  animated,
  bordered,
  onClick
}) => {
  const [showMore, setShowMore] = useState(false);
  return /*#__PURE__*/_jsxs(PaginationItem, {
    value: value,
    onlyDots: onlyDots,
    animated: animated,
    bordered: bordered,
    onClick: e => onClick && onClick(e),
    onMouseEnter: () => setShowMore(true),
    onMouseLeave: () => setShowMore(false),
    children: [showMore ? /*#__PURE__*/_jsxs("svg", {
      role: "presentation",
      focusable: "false",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      strokeWidth: "1.5",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      fill: "none",
      shapeRendering: "geometricPrecision",
      className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]]) + " " + "more",
      children: [/*#__PURE__*/_jsx("path", {
        d: "M13 17l5-5-5-5",
        className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]])
      }), /*#__PURE__*/_jsx("path", {
        d: "M6 17l5-5-5-5",
        className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]])
      })]
    }) : /*#__PURE__*/_jsxs("svg", {
      viewBox: "0 0 24 24",
      strokeWidth: "1.5",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      fill: "none",
      shapeRendering: "geometricPrecision",
      className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]]),
      children: [/*#__PURE__*/_jsx("circle", {
        cx: "12",
        cy: "12",
        r: "1",
        fill: "currentColor",
        className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]])
      }), /*#__PURE__*/_jsx("circle", {
        cx: "19",
        cy: "12",
        r: "1",
        fill: "currentColor",
        className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]])
      }), /*#__PURE__*/_jsx("circle", {
        cx: "5",
        cy: "12",
        r: "1",
        fill: "currentColor",
        className: _JSXStyle.dynamic([["15509035", [isBefore ? '180deg' : '0deg']]])
      })]
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "15509035",
      dynamic: [isBefore ? '180deg' : '0deg'],
      children: `svg.__jsx-style-dynamic-selector{color:currentColor;stroke:currentColor;}.more.__jsx-style-dynamic-selector{-webkit-transform:rotate(${isBefore ? '180deg' : '0deg'});-ms-transform:rotate(${isBefore ? '180deg' : '0deg'});transform:rotate(${isBefore ? '180deg' : '0deg'});}`
    })]
  });
};

export default PaginationEllipsis;
