// @nextui-org/react/esm/pagination/pagination-item.js
import _JSXStyle from "styled-jsx/style";
import React, { useMemo } from 'react';
import useTheme from '../use-theme';
import { addColorAlpha } from '../utils/color';
import { DOTS } from '../use-pagination';
import clsx from '../utils/clsx';
import withDefaults from '../utils/with-defaults';
import { getFocusStyles } from '../utils/styles';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  preserveContent: false
};

const getItemAriaLabel = page => {
  if (!page) return;

  switch (page) {
    case DOTS:
      return 'dots element';

    case '<':
      return 'previous page button';

    case '>':
      return 'next page button';

    case 'first':
      return 'first page button';

    case 'last':
      return 'last page button';

    default:
      return `${page} item`;
  }
};

const PaginationItem = ({
  active,
  value,
  children,
  disabled,
  animated,
  bordered,
  onClick,
  onlyDots,
  preserveContent,
  ...props
}) => {
  const theme = useTheme();
  const ariaLabel = useMemo(() => active ? `${getItemAriaLabel(value)} active` : getItemAriaLabel(value), [value, active]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = getFocusStyles(theme);
  const [hover] = useMemo(() => [addColorAlpha(theme.palette.accents_3, 0.2)], [theme.palette.primary]);

  const clickHandler = event => {
    if (disabled) return;
    onClick && onClick(event);
  };

  return /*#__PURE__*/_jsxs("button", {
    onClick: clickHandler,
    "aria-label": ariaLabel,
    tabIndex: disabled ? -1 : 0,
    ...props,
    className: _JSXStyle.dynamic([["3128098789", [theme.palette.text, theme.palette.accents_1, animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none', hover, theme.expressiveness.shadowSmall, theme.palette.accents_4, theme.palette.accents_2]]]) + " " + (props && props.className != null && props.className || clsx({
      active,
      animated,
      disabled,
      bordered,
      'only-dots': onlyDots,
      'preserve-content': preserveContent
    }, focusClassName) || ""),
    children: [/*#__PURE__*/_jsx("span", {
      className: _JSXStyle.dynamic([["3128098789", [theme.palette.text, theme.palette.accents_1, animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none', hover, theme.expressiveness.shadowSmall, theme.palette.accents_4, theme.palette.accents_2]]]) + " " + "button-content",
      children: children
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "3128098789",
      dynamic: [theme.palette.text, theme.palette.accents_1, animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none', hover, theme.expressiveness.shadowSmall, theme.palette.accents_4, theme.palette.accents_2],
      children: `button.__jsx-style-dynamic-selector{border:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin:0 var(--next-ui-pagination-item-margin);-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0;box-sizing:border-box;text-transform:capitalize;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;text-align:center;vertical-align:middle;box-shadow:none;outline:none;height:var(--next-ui-pagination-size);min-width:var(--next-ui-pagination-size);font-size:inherit;cursor:pointer;border-radius:var(--next-ui-pagination-item-radius);color:${theme.palette.text};background-color:${theme.palette.accents_1};-webkit-transition:${animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none'};transition:${animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none'};}.button-content.__jsx-style-dynamic-selector{position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;top:0;left:0;z-index:20;}.only-dots.__jsx-style-dynamic-selector:not(.preserve-content) .button-content.__jsx-style-dynamic-selector{display:none;}button.__jsx-style-dynamic-selector:hover{background:${hover};}.animated.__jsx-style-dynamic-selector:not(.disabled):not(.active):active{-webkit-transform:scale(var(--next-ui-pagination-scale-transform));-ms-transform:scale(var(--next-ui-pagination-scale-transform));transform:scale(var(--next-ui-pagination-scale-transform));font-size:calc(var(--next-ui-pagination-font-size) * 0.9);}.active.__jsx-style-dynamic-selector{font-weight:bold;cursor:default;box-shadow:${theme.expressiveness.shadowSmall};}.active.__jsx-style-dynamic-selector .button-content.__jsx-style-dynamic-selector{color:white;}.disabled.__jsx-style-dynamic-selector{color:${theme.palette.accents_4};cursor:not-allowed;}.bordered.__jsx-style-dynamic-selector{background-color:transparent;border:var(--next-ui-pagination-item-border-weight) solid ${theme.palette.accents_2};}button.__jsx-style-dynamic-selector svg{width:var(--next-ui-pagination-font-size);height:var(--next-ui-pagination-font-size);}`
    }), focusStyles]
  });
};

export default withDefaults(PaginationItem, defaultProps);
