// @nextui-org/react/esm/modal/modal-footer.js
import _JSXStyle from "styled-jsx/style";
import React, { useMemo, useContext } from 'react';
import useTheme from '../use-theme';
import withDefaults from '../utils/with-defaults';
import { ModalContext } from './modal-context';
import cslx from '../utils/clsx';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  className: '',
  justify: 'flex-end',
  autoMargin: true
};

const ModalFooter = ({
  children,
  className,
  justify,
  autoMargin: autoMarginProp,
  ...props
}) => {
  const theme = useTheme();
  const {
    autoMargin: autoMarginContext,
    noPadding
  } = useContext(ModalContext);
  const autoMargin = useMemo(() => {
    return autoMarginContext !== undefined ? autoMarginContext : autoMarginProp;
  }, [autoMarginProp, autoMarginContext]);
  return /*#__PURE__*/_jsxs("div", { ...props,
    className: _JSXStyle.dynamic([["4275479145", [theme.layout.gapHalf, theme.layout.gap, justify]]]) + " " + (props && props.className != null && props.className || cslx('modal-footer', {
      'auto-margin': autoMargin,
      'no-padding': noPadding
    }, className) || ""),
    children: [children, /*#__PURE__*/_jsx(_JSXStyle, {
      id: "4275479145",
      dynamic: [theme.layout.gapHalf, theme.layout.gap, justify],
      children: `.modal-footer.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;overflow:hidden;color:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;font-size:0.875rem;padding:${theme.layout.gapHalf} ${theme.layout.gap};-webkit-box-pack:${justify};-webkit-justify-content:${justify};-ms-flex-pack:${justify};justify-content:${justify};}.no-padding.__jsx-style-dynamic-selector{padding:0;}.auto-margin.__jsx-style-dynamic-selector>*{margin:0.25rem;}`
    })]
  });
};

const MemoModalFooter = /*#__PURE__*/React.memo(ModalFooter);
export default withDefaults(MemoModalFooter, defaultProps);
