// @nextui-org/react/esm/card/styles.js
import { getNormalColor, hexFromString, addColorAlpha } from './../utils/color';
export const getDripColor = (palette, color, bordered) => {
  const colors = {
    default: palette.accents_2,
    primary: palette.primary,
    secondary: palette.secondary,
    success: palette.success,
    warning: palette.warning,
    error: palette.error,
    gradient: hexFromString(palette.gradient, palette.primary, true)
  };
  const baseColor = color === 'default' ? palette.primary : colors[color || 'default'] || getNormalColor(color, palette);
  const selectedColor = bordered ? baseColor : palette.accents_6;
  if (selectedColor) return addColorAlpha(selectedColor, 0.25);
  return palette.accents_2;
};
export const getStyles = (color, textColor, shadow, bordered, theme) => {
  const palette = theme.palette;
  const isDark = theme.type === 'dark';
  const normalColor = getNormalColor(color, palette, isDark ? palette.accents_1 : palette.background);
  const normalTextColor = getNormalColor(textColor, palette, palette.text);
  const dripColor = getDripColor(palette, color, bordered);
  return {
    bgColor: color === 'default' && !shadow && !bordered ? palette.accents_1 : normalColor,
    color: normalTextColor,
    borderColor: color === 'default' ? palette.border : normalColor,
    dripColor
  };
};
