// @nextui-org/react/esm/col/col.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import withDefaults from '../utils/with-defaults';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  span: 12,
  offset: 0,
  as: 'div',
  className: ''
};

const Col = ({
  as,
  children,
  span,
  offset,
  className,
  ...props
}) => {
  const Component = as;
  return /*#__PURE__*/_jsxs(Component, { ...props,
    className: _JSXStyle.dynamic([["2614916236", [100 / 12 * span, 100 / 12 * offset]]]) + " " + (props && props.className != null && props.className || `col ${className}`),
    children: [children, /*#__PURE__*/_jsx(_JSXStyle, {
      id: "2614916236",
      dynamic: [100 / 12 * span, 100 / 12 * offset],
      children: `.col.__jsx-style-dynamic-selector{float:left;box-sizing:border-box;padding-left:calc(var(--row-gap) / 2);padding-right:calc(var(--row-gap) / 2);width:${100 / 12 * span}%;margin-left:${100 / 12 * offset}%;}`
    })]
  });
};

const MemoCol = /*#__PURE__*/React.memo(Col);
export default withDefaults(MemoCol, defaultProps);
