// @nextui-org/react/esm/user/user-link.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import Link from '../link';
import withDefaults from '../utils/with-defaults';
import { __DEV__ } from '../utils/assertion';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  className: ''
};
const UserLink = /*#__PURE__*/React.forwardRef(({
  href,
  className,
  children,
  ...props
}, ref) => {
  return /*#__PURE__*/_jsxs("div", { ...props,
    className: "jsx-3263947648" + " " + (props && props.className != null && props.className || className || ""),
    children: [/*#__PURE__*/_jsx(Link, {
      ref: ref,
      href: href,
      color: true,
      target: "_blank",
      rel: "noopener",
      children: children
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "3263947648",
      children: "div.jsx-3263947648 a:hover{opacity:0.7;}"
    })]
  });
});

if (__DEV__) {
  UserLink.displayName = 'UserLink';
}

const MemoUserLink = /*#__PURE__*/React.memo(UserLink);
export default withDefaults(MemoUserLink, defaultProps);
