// @nextui-org/react/esm/user/user.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import { Avatar } from '../index';
import useTheme from '../use-theme';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  className: '',
  size: 32,
  squared: false,
  bordered: false
};

const User = ({
  src,
  text,
  name,
  children,
  className,
  altText,
  color,
  squared,
  bordered,
  size,
  ...props
}) => {
  const theme = useTheme();
  return /*#__PURE__*/_jsxs("div", { ...props,
    className: _JSXStyle.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + (props && props.className != null && props.className || `user ${className}`),
    children: [/*#__PURE__*/_jsx(Avatar, {
      src: src,
      color: color,
      squared: squared,
      bordered: bordered,
      text: text,
      size: size,
      alt: altText
    }), /*#__PURE__*/_jsxs("div", {
      className: _JSXStyle.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + "names",
      children: [/*#__PURE__*/_jsx("span", {
        className: _JSXStyle.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + "name",
        children: name
      }), /*#__PURE__*/_jsx("span", {
        className: _JSXStyle.dynamic([["1525555450", [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4]]]) + " " + "social",
        children: children
      })]
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "1525555450",
      dynamic: [theme.layout.gapHalf, theme.layout.gapHalf, theme.palette.text, theme.palette.accents_4],
      children: `.user.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;padding:0 ${theme.layout.gapHalf};-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:100%;}.names.__jsx-style-dynamic-selector{margin-left:${theme.layout.gapHalf};display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;white-space:nowrap;}.name.__jsx-style-dynamic-selector{font-size:0.89rem;color:${theme.palette.text};line-height:1.1rem;text-transform:capitalize;font-weight:500;max-width:15rem;text-overflow:ellipsis;overflow:hidden;}.social.__jsx-style-dynamic-selector{font-size:0.75rem;color:${theme.palette.accents_4};}.social.__jsx-style-dynamic-selector *:first-child{margin-top:0;}.social.__jsx-style-dynamic-selector *:last-child{margin-bottom:0;}`
    })]
  });
};

User.defaultProps = defaultProps;
export default /*#__PURE__*/React.memo(User);
