// @nextui-org/react/cjs/col/col.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireDefault(require("react"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _jsxRuntime = require("react/jsx-runtime");

const defaultProps = {
  span: 12,
  offset: 0,
  as: 'div',
  className: ''
};

const Col = ({
  as,
  children,
  span,
  offset,
  className,
  ...props
}) => {
  const Component = as;
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(Component, { ...props,
    className: _style.default.dynamic([["2614916236", [100 / 12 * span, 100 / 12 * offset]]]) + " " + (props && props.className != null && props.className || `col ${className}`),
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "2614916236",
      dynamic: [100 / 12 * span, 100 / 12 * offset],
      children: `.col.__jsx-style-dynamic-selector{float:left;box-sizing:border-box;padding-left:calc(var(--row-gap) / 2);padding-right:calc(var(--row-gap) / 2);width:${100 / 12 * span}%;margin-left:${100 / 12 * offset}%;}`
    })]
  });
};

const MemoCol = /*#__PURE__*/_react.default.memo(Col);

var _default = (0, _withDefaults.default)(MemoCol, defaultProps);

exports.default = _default;
module.exports = exports.default;
