// @nextui-org/react/esm/button/button-loading.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import withDefaults from '../utils/with-defaults';
import Loading from '../loading';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  size: 'small',
  type: 'default'
};

const ButtonLoading = ({
  color,
  size,
  type,
  background,
  ...props
}) => {
  return /*#__PURE__*/_jsxs("div", { ...props,
    className: "jsx-3876315024" + " " + (props && props.className != null && props.className || "button-loading"),
    children: [/*#__PURE__*/_jsx(Loading, {
      size: size,
      color: color,
      gradientBackground: background,
      type: type
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "3876315024",
      children: ".button-loading.jsx-3876315024{top:0;left:0;right:0;bottom:0;z-index:2;position:absolute;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;border-radius:var(--next-ui-button-border-radius);background-color:var(--next-ui-button-bg);}"
    })]
  });
};

const MemoButtonLoading = /*#__PURE__*/React.memo(ButtonLoading);
export default withDefaults(MemoButtonLoading, defaultProps);
