// @nextui-org/react/esm/button/button.js
import _JSXStyle from "styled-jsx/style";
import React, { useRef, useMemo, useImperativeHandle } from 'react';
import useWarning from '../use-warning';
import useTheme from '../use-theme';
import ButtonDrip from '../utils/drip';
import ButtonLoading from './button-loading';
import { filterPropsWithGroup } from './utils';
import { useButtonGroupContext } from './button-group-context';
import ButtonIcon from './button-icon';
import { getButtonColors, getButtonCursor, getButtonDripColor, getButtonSize } from './styles';
import { getNormalShadowColor } from '../utils/color';
import { getFocusStyles } from '../utils/styles';
import { getNormalRadius } from '../utils/dimensions';
import { __DEV__ } from '../utils/assertion';
import clsx from '../utils/clsx';
import useDrip from '../use-drip';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  color: 'default',
  size: 'medium',
  htmlType: 'button',
  loaderType: 'default',
  borderWeight: 'normal',
  flat: false,
  light: false,
  loading: false,
  rounded: false,
  shadow: false,
  auto: false,
  bordered: false,
  ghost: false,
  animated: true,
  disabled: false,
  className: ''
};
const Button = /*#__PURE__*/React.forwardRef(({ ...btnProps
}, ref) => {
  var _hover$style, _hover$border, _hover$border2, _hover$style2, _hover$style3, _hover$border3, _hover$border4;

  const theme = useTheme();
  const buttonRef = useRef(null);
  useImperativeHandle(ref, () => buttonRef.current);
  const groupConfig = useButtonGroupContext();
  const filteredProps = filterPropsWithGroup(btnProps, groupConfig);
  /* eslint-disable @typescript-eslint/no-unused-vars */

  const {
    children,
    disabled,
    loading,
    shadow,
    animated,
    light,
    flat,
    rounded,
    onClick,
    auto,
    size,
    icon,
    htmlType,
    iconRight,
    className,
    loaderType,
    bordered,
    ghost,
    style: buttonStyle,
    ...props
  } = filteredProps;
  /* eslint-enable @typescript-eslint/no-unused-vars */

  if (__DEV__ && filteredProps.color === 'gradient' && (flat || light)) {
    useWarning('Using the gradient color on flat and light buttons will have no effect.');
  }

  const hasIcon = icon || iconRight;
  const isRight = Boolean(iconRight);
  const {
    bg,
    color,
    loaderBg,
    border,
    style,
    hover
  } = useMemo(() => getButtonColors(theme.palette, filteredProps), [theme.palette, filteredProps]);
  const radius = useMemo(() => getNormalRadius(size, rounded), [size, rounded]);
  const shadowColor = useMemo(() => shadow ? getNormalShadowColor(filteredProps.color, theme.palette) : '', [theme.palette, filteredProps, shadow]);
  const {
    cursor,
    events
  } = useMemo(() => getButtonCursor(disabled, loading), [disabled, loading]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = getFocusStyles(theme);
  const {
    height,
    minWidth,
    padding,
    width,
    fontSize,
    loaderSize
  } = useMemo(() => getButtonSize(size, auto), [size, auto]);
  const dripColor = useMemo(() => getButtonDripColor(theme.palette, filteredProps), [theme.palette, filteredProps]);
  const {
    onClick: onDripClickHandler,
    ...dripBindings
  } = useDrip(false, buttonRef);
  const paddingForAutoMode = useMemo(() => auto || size === 'mini' ? `calc(var(--next-ui-button-height) / 2 + var(--next-ui-button-padding) * .5)` : 0, [auto, size]);
  const hoverBeforeOpacity = useMemo(() => filteredProps.color === 'gradient' && ghost ? 1 : 0, [ghost, filteredProps]);
  const paddingForBorderedGradient = useMemo(() => filteredProps.color === 'gradient' && (bordered || ghost) ? `var(--next-ui-button-padding)` : 0, [filteredProps.color, bordered]);

  const clickHandler = event => {
    if (disabled || loading) return;

    if (animated && buttonRef.current) {
      onDripClickHandler(event);
    }

    onClick && onClick(event);
  }; // to avoid passing borderweight prop to the html button element


  delete props.borderWeight;
  return /*#__PURE__*/_jsxs("button", {
    ref: buttonRef,
    type: htmlType,
    disabled: disabled,
    onClick: clickHandler,
    style: { ...style,
      ...buttonStyle
    },
    ...props,
    className: _JSXStyle.dynamic([["2827454966", [bg, padding, height, height, minWidth, width, radius, (border == null ? void 0 : border.width) || '2px', (border == null ? void 0 : border.display) || 'none', (border == null ? void 0 : border.color) || 'transparent', fontSize, color, cursor, events, shadowColor, radius, padding, height, color, bg, hover == null ? void 0 : hover.bg, radius, hoverBeforeOpacity, animated ? 'scale(0.97)' : 'none', (hover == null ? void 0 : hover.bg) || 'inherit', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$style = hover.style) == null ? void 0 : _hover$style.filter) || 'none', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$border = hover.border) == null ? void 0 : _hover$border.color) || 'transparent', hover == null ? void 0 : (_hover$border2 = hover.border) == null ? void 0 : _hover$border2.width, hover == null ? void 0 : hover.padding, cursor, events, bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0', paddingForBorderedGradient, loading ? 0 : 1, paddingForAutoMode, paddingForAutoMode]]]) + " " + (props && props.className != null && props.className || clsx('button', {
      disabled
    }, focusClassName, className) || ""),
    children: [loading && /*#__PURE__*/_jsx(ButtonLoading, {
      size: loaderSize,
      type: loaderType,
      color: color,
      background: loaderBg
    }), React.Children.count(children) === 0 ? /*#__PURE__*/_jsx(ButtonIcon, {
      isRight: isRight,
      isSingle: true,
      children: hasIcon
    }) : hasIcon ? /*#__PURE__*/_jsxs("div", {
      className: _JSXStyle.dynamic([["2827454966", [bg, padding, height, height, minWidth, width, radius, (border == null ? void 0 : border.width) || '2px', (border == null ? void 0 : border.display) || 'none', (border == null ? void 0 : border.color) || 'transparent', fontSize, color, cursor, events, shadowColor, radius, padding, height, color, bg, hover == null ? void 0 : hover.bg, radius, hoverBeforeOpacity, animated ? 'scale(0.97)' : 'none', (hover == null ? void 0 : hover.bg) || 'inherit', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$style = hover.style) == null ? void 0 : _hover$style.filter) || 'none', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$border = hover.border) == null ? void 0 : _hover$border.color) || 'transparent', hover == null ? void 0 : (_hover$border2 = hover.border) == null ? void 0 : _hover$border2.width, hover == null ? void 0 : hover.padding, cursor, events, bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0', paddingForBorderedGradient, loading ? 0 : 1, paddingForAutoMode, paddingForAutoMode]]]),
      children: [/*#__PURE__*/_jsx(ButtonIcon, {
        isRight: isRight,
        children: hasIcon
      }), /*#__PURE__*/_jsx("div", {
        className: _JSXStyle.dynamic([["2827454966", [bg, padding, height, height, minWidth, width, radius, (border == null ? void 0 : border.width) || '2px', (border == null ? void 0 : border.display) || 'none', (border == null ? void 0 : border.color) || 'transparent', fontSize, color, cursor, events, shadowColor, radius, padding, height, color, bg, hover == null ? void 0 : hover.bg, radius, hoverBeforeOpacity, animated ? 'scale(0.97)' : 'none', (hover == null ? void 0 : hover.bg) || 'inherit', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$style = hover.style) == null ? void 0 : _hover$style.filter) || 'none', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$border = hover.border) == null ? void 0 : _hover$border.color) || 'transparent', hover == null ? void 0 : (_hover$border2 = hover.border) == null ? void 0 : _hover$border2.width, hover == null ? void 0 : hover.padding, cursor, events, bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0', paddingForBorderedGradient, loading ? 0 : 1, paddingForAutoMode, paddingForAutoMode]]]) + " " + `text ${isRight ? 'right' : 'left'}`,
        children: children
      })]
    }) : /*#__PURE__*/_jsx("div", {
      className: _JSXStyle.dynamic([["2827454966", [bg, padding, height, height, minWidth, width, radius, (border == null ? void 0 : border.width) || '2px', (border == null ? void 0 : border.display) || 'none', (border == null ? void 0 : border.color) || 'transparent', fontSize, color, cursor, events, shadowColor, radius, padding, height, color, bg, hover == null ? void 0 : hover.bg, radius, hoverBeforeOpacity, animated ? 'scale(0.97)' : 'none', (hover == null ? void 0 : hover.bg) || 'inherit', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$style = hover.style) == null ? void 0 : _hover$style.filter) || 'none', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$border = hover.border) == null ? void 0 : _hover$border.color) || 'transparent', hover == null ? void 0 : (_hover$border2 = hover.border) == null ? void 0 : _hover$border2.width, hover == null ? void 0 : hover.padding, cursor, events, bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0', paddingForBorderedGradient, loading ? 0 : 1, paddingForAutoMode, paddingForAutoMode]]]) + " " + "text",
      children: children
    }), /*#__PURE__*/_jsx(ButtonDrip, {
      color: dripColor,
      ...dripBindings
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "2827454966",
      dynamic: [bg, padding, height, height, minWidth, width, radius, (border == null ? void 0 : border.width) || '2px', (border == null ? void 0 : border.display) || 'none', (border == null ? void 0 : border.color) || 'transparent', fontSize, color, cursor, events, shadowColor, radius, padding, height, color, bg, hover == null ? void 0 : hover.bg, radius, hoverBeforeOpacity, animated ? 'scale(0.97)' : 'none', (hover == null ? void 0 : hover.bg) || 'inherit', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$style = hover.style) == null ? void 0 : _hover$style.filter) || 'none', hover == null ? void 0 : hover.color, (hover == null ? void 0 : (_hover$border = hover.border) == null ? void 0 : _hover$border.color) || 'transparent', hover == null ? void 0 : (_hover$border2 = hover.border) == null ? void 0 : _hover$border2.width, hover == null ? void 0 : hover.padding, cursor, events, bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0', paddingForBorderedGradient, loading ? 0 : 1, paddingForAutoMode, paddingForAutoMode],
      children: `.button.__jsx-style-dynamic-selector{background:${bg};box-sizing:border-box;display:inline-block;padding:0 ${padding};height:${height};line-height:${height};min-width:${minWidth};width:${width};border-radius:${radius};border:${(border == null ? void 0 : border.width) || '2px'} ${(border == null ? void 0 : border.display) || 'none'} ${(border == null ? void 0 : border.color) || 'transparent'};font-weight:500;font-size:${fontSize};-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;text-transform:capitalize;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;text-align:center;white-space:nowrap;-webkit-transition:all 250ms ease;transition:all 250ms ease;position:relative;overflow:hidden;color:${color};cursor:${cursor};pointer-events:${events};box-shadow:${shadowColor};--next-ui-button-border-radius:${radius};--next-ui-button-padding:${padding};--next-ui-button-height:${height};--next-ui-button-color:${color};--next-ui-button-bg:${bg};}.button.__jsx-style-dynamic-selector:before{content:'';background:${hover == null ? void 0 : hover.bg};position:absolute;top:0;left:0;height:100%;width:100%;border-radius:${radius};-webkit-transition:all 0.2s ease;transition:all 0.2s ease;z-index:1;opacity:0;}.button.__jsx-style-dynamic-selector:hover.__jsx-style-dynamic-selector:before{opacity:${hoverBeforeOpacity};}.button.__jsx-style-dynamic-selector:not(.disabled):active{-webkit-transform:${animated ? 'scale(0.97)' : 'none'};-ms-transform:${animated ? 'scale(0.97)' : 'none'};transform:${animated ? 'scale(0.97)' : 'none'};}.button.__jsx-style-dynamic-selector:hover,.button.__jsx-style-dynamic-selector:focus{background:${(hover == null ? void 0 : hover.bg) || 'inherit'};--next-ui-button-color:${hover == null ? void 0 : hover.color};-webkit-filter:${(hover == null ? void 0 : (_hover$style2 = hover.style) == null ? void 0 : _hover$style2.filter) || 'none'};filter:${(hover == null ? void 0 : (_hover$style3 = hover.style) == null ? void 0 : _hover$style3.filter) || 'none'};color:${hover == null ? void 0 : hover.color};border-color:${(hover == null ? void 0 : (_hover$border3 = hover.border) == null ? void 0 : _hover$border3.color) || 'transparent'};border-width:${hover == null ? void 0 : (_hover$border4 = hover.border) == null ? void 0 : _hover$border4.width};padding:${hover == null ? void 0 : hover.padding} !important;cursor:${cursor};pointer-events:${events};}.text.__jsx-style-dynamic-selector{position:relative;z-index:1;-webkit-transform:translateY( -${bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0'} );-ms-transform:translateY( -${bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0'} );transform:translateY( -${bordered || ghost ? (border == null ? void 0 : border.width) || '2px' : '0'} );display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;text-align:center;line-height:inherit;padding:0 ${paddingForBorderedGradient};opacity:${loading ? 0 : 1};}.text.__jsx-style-dynamic-selector p.__jsx-style-dynamic-selector,.text.__jsx-style-dynamic-selector pre.__jsx-style-dynamic-selector,.text.__jsx-style-dynamic-selector div.__jsx-style-dynamic-selector{margin:0;}.text.left.__jsx-style-dynamic-selector{padding-left:${paddingForAutoMode};}.text.right.__jsx-style-dynamic-selector{padding-right:${paddingForAutoMode};}`
    }), focusStyles]
  });
});

if (__DEV__) {
  Button.displayName = 'NextUI - Button';
}

Button.defaultProps = defaultProps;
export default Button;
