// @nextui-org/react/cjs/button/button-loading.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireDefault(require("react"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _loading = _interopRequireDefault(require("../loading"));

var _jsxRuntime = require("react/jsx-runtime");

const defaultProps = {
  size: 'small',
  type: 'default'
};

const ButtonLoading = ({
  color,
  size,
  type,
  background,
  ...props
}) => {
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", { ...props,
    className: "jsx-3876315024" + " " + (props && props.className != null && props.className || "button-loading"),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_loading.default, {
      size: size,
      color: color,
      gradientBackground: background,
      type: type
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3876315024",
      children: ".button-loading.jsx-3876315024{top:0;left:0;right:0;bottom:0;z-index:2;position:absolute;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;border-radius:var(--next-ui-button-border-radius);background-color:var(--next-ui-button-bg);}"
    })]
  });
};

const MemoButtonLoading = /*#__PURE__*/_react.default.memo(ButtonLoading);

var _default = (0, _withDefaults.default)(MemoButtonLoading, defaultProps);

exports.default = _default;
module.exports = exports.default;
