// @nextui-org/react/cjs/button/button-group.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _buttonGroupContext = require("./button-group-context");

var _styles = require("./styles");

var _dimensions = require("../utils/dimensions");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  disabled: false,
  vertical: false,
  bordered: false,
  light: false,
  flat: false,
  shadow: false,
  auto: false,
  animated: false,
  rounded: false,
  ghost: false,
  borderWeight: 'normal',
  size: 'medium',
  color: 'default',
  className: ''
};

const ButtonGroup = groupProps => {
  const theme = (0, _useTheme.default)();
  const {
    disabled,
    size,
    color,
    bordered,
    ghost,
    light,
    flat,
    shadow,
    auto,
    animated,
    rounded,
    vertical,
    children,
    className,
    ...props
  } = groupProps;
  const initialValue = (0, _react.useMemo)(() => ({
    disabled,
    size,
    color,
    bordered,
    light,
    ghost,
    flat,
    shadow,
    auto,
    animated,
    rounded,
    isButtonGroup: true
  }), [disabled, size, color, bordered, light, ghost, flat]);
  const {
    color: borderColor,
    width: borderWidth
  } = (0, _react.useMemo)(() => {
    return (0, _styles.getGroupBorder)(theme.palette, groupProps);
  }, [theme, disabled, bordered]);
  const radius = (0, _react.useMemo)(() => (0, _dimensions.getNormalRadius)(size, rounded), [size, rounded]); // to avoid passing borderweight prop to the html button element

  delete props.borderWeight;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_buttonGroupContext.ButtonGroupContext.Provider, {
    value: initialValue,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", { ...props,
      className: _style.default.dynamic([["2062957038", [radius, theme.layout.gapQuarter, borderWidth, borderColor, borderWidth, borderColor, borderWidth, borderColor]]]) + " " + (props && props.className != null && props.className || `button-group ${vertical ? 'vertical' : 'horizontal'} ${groupProps.color === 'gradient' ? 'gradient' : ''} ${className}`),
      children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
        id: "2062957038",
        dynamic: [radius, theme.layout.gapQuarter, borderWidth, borderColor, borderWidth, borderColor, borderWidth, borderColor],
        children: `.button-group.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;border-radius:${radius};margin:${theme.layout.gapQuarter};border:${borderWidth} solid ${borderColor};background-color:transparent;height:-webkit-min-content;height:-moz-min-content;height:min-content;}.vertical.__jsx-style-dynamic-selector{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.button-group.__jsx-style-dynamic-selector .button .text{top:0;}.button-group.__jsx-style-dynamic-selector .button{border:none;}.horizontal.__jsx-style-dynamic-selector .button:not(:first-child){border-top-left-radius:0;border-bottom-left-radius:0;border-left:${borderWidth} solid ${borderColor};}.horizontal.__jsx-style-dynamic-selector .button:not(:first-child):before{border-radius:0;}.horizontal.__jsx-style-dynamic-selector .button:not(:last-child){border-top-right-radius:0;border-bottom-right-radius:0;}.horizontal.__jsx-style-dynamic-selector .button:not(:last-child):before{border-radius:0;}.gradient.horizontal.__jsx-style-dynamic-selector .button:not(:last-child):not(:first-child){padding-left:0 !important;-webkit-filter:hue-rotate(310deg);filter:hue-rotate(310deg);}.gradient.horizontal.__jsx-style-dynamic-selector .button:last-child{-webkit-filter:hue-rotate(250deg);filter:hue-rotate(250deg);padding-left:0 !important;}.gradient.vertical.__jsx-style-dynamic-selector .button:not(:last-child):not(:first-child){padding-top:0 !important;}.gradient.vertical.__jsx-style-dynamic-selector .button:last-child{padding-top:0 !important;}.vertical.__jsx-style-dynamic-selector .button:not(:first-child){border-top-left-radius:0;border-top-right-radius:0;border-top:${borderWidth} solid ${borderColor};}.vertical.__jsx-style-dynamic-selector .button:not(:first-child):before{border-radius:0;}.vertical.__jsx-style-dynamic-selector .button:not(:last-child){border-bottom-left-radius:0;border-bottom-right-radius:0;}.vertical.__jsx-style-dynamic-selector .button:not(:last-child):before{border-bottom-left-radius:0;border-bottom-right-radius:0;}`
      })]
    })
  });
};

const MemoButtonGroup = /*#__PURE__*/_react.default.memo(ButtonGroup);

var _default = (0, _withDefaults.default)(MemoButtonGroup, defaultProps);

exports.default = _default;
module.exports = exports.default;
