// @nextui-org/react/cjs/progress/progress.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _cssTransition = _interopRequireDefault(require("../utils/css-transition"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _color = require("../utils/color");

var _styles = require("./styles");

var _numbers = require("../utils/numbers");

var _styles2 = require("../utils/styles");

var _clsx = _interopRequireDefault(require("../utils/clsx"));

var _assertion = require("../utils/assertion");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  color: 'primary',
  status: 'default',
  size: 'medium',
  striped: false,
  animated: true,
  squared: false,
  shadow: false,
  indeterminated: false,
  value: 0,
  min: 0,
  max: 100,
  className: ''
};

const ProgressBar = ({
  color,
  value: valueProp,
  max,
  min,
  striped,
  status,
  animated,
  shadow,
  squared,
  size,
  indeterminated,
  className,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  const value = (0, _react.useMemo)(() => valueProp > max ? max : valueProp < min ? min : valueProp, [valueProp, min, max]);
  const percent = (0, _react.useMemo)(() => (0, _numbers.valueToPercent)(value, min, max), [value, min, max]);
  const fillerColor = (0, _react.useMemo)(() => (0, _color.getNormalColor)(color, theme.palette, theme.palette.primary), [color, theme.palette]);
  const shadowColor = (0, _react.useMemo)(() => shadow ? (0, _styles.getShadowColor)(color, theme.palette) : 'none', [color, shadow, theme.palette]);
  const {
    height,
    radius
  } = (0, _react.useMemo)(() => (0, _styles.getSizes)(size, squared), [size, squared]);
  const bgColor = (0, _react.useMemo)(() => {
    if (status === 'default') {
      return theme.palette.accents_2;
    }

    const normalColor = (0, _color.getNormalColor)(status, theme.palette, theme.palette.accents_2);
    return (0, _color.addColorAlpha)(normalColor, 0.2);
  }, [status, theme.palette]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = (0, _styles2.getFocusStyles)(theme);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    role: "progressbar",
    ...props,
    className: _style.default.dynamic([["3309033417", [height, !indeterminated ? 'visible' : 'hidden', bgColor, radius, fillerColor, animated ? 'width 0.5s cubic-bezier(0.4, 0, 0.2, 1)' : 'none', shadowColor, percent, theme.layout.gap, theme.layout.gap]]]) + " " + (props && props.className != null && props.className || (0, _clsx.default)('progress', className) || ""),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_cssTransition.default, {
      visible: true,
      name: "progress-wrapper",
      enterTime: 10,
      leaveTime: 20,
      clearTime: 300,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        tabIndex: 0,
        "aria-valuenow": value,
        "aria-valuemin": min,
        "aria-valuemax": max,
        ...props,
        className: _style.default.dynamic([["3309033417", [height, !indeterminated ? 'visible' : 'hidden', bgColor, radius, fillerColor, animated ? 'width 0.5s cubic-bezier(0.4, 0, 0.2, 1)' : 'none', shadowColor, percent, theme.layout.gap, theme.layout.gap]]]) + " " + (props && props.className != null && props.className || (0, _clsx.default)('filler', {
          striped,
          indeterminated
        }, focusClassName) || "")
      })
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3309033417",
      dynamic: [height, !indeterminated ? 'visible' : 'hidden', bgColor, radius, fillerColor, animated ? 'width 0.5s cubic-bezier(0.4, 0, 0.2, 1)' : 'none', shadowColor, percent, theme.layout.gap, theme.layout.gap],
      children: `.progress.__jsx-style-dynamic-selector{margin:0;padding:0;width:100%;height:${height};position:relative;overflow:${!indeterminated ? 'visible' : 'hidden'};background:${bgColor};border-radius:${radius};}.filler.__jsx-style-dynamic-selector{margin:0;padding:0;width:0;opacity:0;height:100%;min-width:inherit;background:${fillerColor};border-radius:inherit;-webkit-transition:${animated ? 'width 0.5s cubic-bezier(0.4, 0, 0.2, 1)' : 'none'};transition:${animated ? 'width 0.5s cubic-bezier(0.4, 0, 0.2, 1)' : 'none'};box-shadow:${shadowColor};-webkit-transition:all 0.25s ease;transition:all 0.25s ease;}.progress-wrapper-enter.__jsx-style-dynamic-selector{opacity:0;}.progress-wrapper-enter-active.__jsx-style-dynamic-selector{opacity:1;width:${percent}%;}.filler.striped.__jsx-style-dynamic-selector{background-image:linear-gradient( 45deg, rgba(0,0,0,0.1) 25%, transparent 25%, transparent 50%, rgba(0,0,0,0.1) 50%, rgba(0,0,0,0.1) 75%, transparent 75%, transparent );background-size:${theme.layout.gap} ${theme.layout.gap};}.filler.indeterminated.__jsx-style-dynamic-selector{position:absolute;width:0%;-webkit-transition-property:background-color,width,left,border-color, opacity,shadow,-webkit-transform;-webkit-transition-property:background-color,width,left,border-color, opacity,shadow,transform;transition-property:background-color,width,left,border-color, opacity,shadow,transform;-webkit-transition-duration:300ms;transition-duration:300ms;will-change:left;min-width:50%;-webkit-animation:1s ease 0s infinite normal none running indeterminate-animation-__jsx-style-dynamic-selector;animation:1s ease 0s infinite normal none running indeterminate-animation-__jsx-style-dynamic-selector;}@-webkit-keyframes indeterminate-animation-__jsx-style-dynamic-selector{0%{left:-40%;}100%{left:100%;}}@keyframes indeterminate-animation-__jsx-style-dynamic-selector{0%{left:-40%;}100%{left:100%;}}`
    }), focusStyles]
  });
};

ProgressBar.defaultProps = defaultProps;

if (_assertion.__DEV__) {
  ProgressBar.displayName = 'NextUI - Progress';
}

var _default = (0, _withDefaults.default)( /*#__PURE__*/_react.default.memo(ProgressBar), defaultProps);

exports.default = _default;
module.exports = exports.default;
