// @nextui-org/react/cjs/progress/styles.js
"use strict";

exports.__esModule = true;
exports.getSizes = exports.getShadowColor = void 0;

var _color = require("../utils/color");

const getShadowColor = (color, palette) => {
  try {
    const hexColor = color === 'gradient' ? (0, _color.hexFromString)(palette.gradient, palette.primary, true) : (0, _color.getNormalColor)(color, palette, palette.primary);
    const [r, g, b] = (0, _color.hexToRgb)(hexColor);
    return `1px 2px 10px 0px rgb(${r} ${g} ${b}/ 60%);`;
  } catch (err) {
    return 'none';
  }
};

exports.getShadowColor = getShadowColor;

const getSizes = (size, squared) => {
  const sizes = {
    mini: {
      height: '0.25rem',
      radius: squared ? '1px' : '0.25rem'
    },
    small: {
      height: '0.5rem',
      radius: squared ? '2px' : '0.5rem'
    },
    medium: {
      height: '1rem',
      radius: squared ? '3px' : '0.75rem'
    },
    large: {
      height: '1.5rem',
      radius: squared ? '4px' : '1rem'
    },
    xlarge: {
      height: '1.75rem',
      radius: squared ? '5px' : '1.5rem'
    }
  };
  if (!size) return sizes.medium;
  return sizes[size];
};

exports.getSizes = getSizes;
