// @nextui-org/react/cjs/pagination/pagination-item.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _color = require("../utils/color");

var _usePagination = require("../use-pagination");

var _clsx = _interopRequireDefault(require("../utils/clsx"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _styles = require("../utils/styles");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  preserveContent: false
};

const getItemAriaLabel = page => {
  if (!page) return;

  switch (page) {
    case _usePagination.DOTS:
      return 'dots element';

    case '<':
      return 'previous page button';

    case '>':
      return 'next page button';

    case 'first':
      return 'first page button';

    case 'last':
      return 'last page button';

    default:
      return `${page} item`;
  }
};

const PaginationItem = ({
  active,
  value,
  children,
  disabled,
  animated,
  bordered,
  onClick,
  onlyDots,
  preserveContent,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  const ariaLabel = (0, _react.useMemo)(() => active ? `${getItemAriaLabel(value)} active` : getItemAriaLabel(value), [value, active]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = (0, _styles.getFocusStyles)(theme);
  const [hover] = (0, _react.useMemo)(() => [(0, _color.addColorAlpha)(theme.palette.accents_3, 0.2)], [theme.palette.primary]);

  const clickHandler = event => {
    if (disabled) return;
    onClick && onClick(event);
  };

  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("button", {
    onClick: clickHandler,
    "aria-label": ariaLabel,
    tabIndex: disabled ? -1 : 0,
    ...props,
    className: _style.default.dynamic([["3128098789", [theme.palette.text, theme.palette.accents_1, animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none', hover, theme.expressiveness.shadowSmall, theme.palette.accents_4, theme.palette.accents_2]]]) + " " + (props && props.className != null && props.className || (0, _clsx.default)({
      active,
      animated,
      disabled,
      bordered,
      'only-dots': onlyDots,
      'preserve-content': preserveContent
    }, focusClassName) || ""),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
      className: _style.default.dynamic([["3128098789", [theme.palette.text, theme.palette.accents_1, animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none', hover, theme.expressiveness.shadowSmall, theme.palette.accents_4, theme.palette.accents_2]]]) + " " + "button-content",
      children: children
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "3128098789",
      dynamic: [theme.palette.text, theme.palette.accents_1, animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none', hover, theme.expressiveness.shadowSmall, theme.palette.accents_4, theme.palette.accents_2],
      children: `button.__jsx-style-dynamic-selector{border:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin:0 var(--next-ui-pagination-item-margin);-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0;box-sizing:border-box;text-transform:capitalize;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;text-align:center;vertical-align:middle;box-shadow:none;outline:none;height:var(--next-ui-pagination-size);min-width:var(--next-ui-pagination-size);font-size:inherit;cursor:pointer;border-radius:var(--next-ui-pagination-item-radius);color:${theme.palette.text};background-color:${theme.palette.accents_1};-webkit-transition:${animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none'};transition:${animated ? 'transform 0.25s ease 0s, background 0.25s ease 0s, box-shadow 0.25s ease 0s' : 'none'};}.button-content.__jsx-style-dynamic-selector{position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;top:0;left:0;z-index:20;}.only-dots.__jsx-style-dynamic-selector:not(.preserve-content) .button-content.__jsx-style-dynamic-selector{display:none;}button.__jsx-style-dynamic-selector:hover{background:${hover};}.animated.__jsx-style-dynamic-selector:not(.disabled):not(.active):active{-webkit-transform:scale(var(--next-ui-pagination-scale-transform));-ms-transform:scale(var(--next-ui-pagination-scale-transform));transform:scale(var(--next-ui-pagination-scale-transform));font-size:calc(var(--next-ui-pagination-font-size) * 0.9);}.active.__jsx-style-dynamic-selector{font-weight:bold;cursor:default;box-shadow:${theme.expressiveness.shadowSmall};}.active.__jsx-style-dynamic-selector .button-content.__jsx-style-dynamic-selector{color:white;}.disabled.__jsx-style-dynamic-selector{color:${theme.palette.accents_4};cursor:not-allowed;}.bordered.__jsx-style-dynamic-selector{background-color:transparent;border:var(--next-ui-pagination-item-border-weight) solid ${theme.palette.accents_2};}button.__jsx-style-dynamic-selector svg{width:var(--next-ui-pagination-font-size);height:var(--next-ui-pagination-font-size);}`
    }), focusStyles]
  });
};

var _default = (0, _withDefaults.default)(PaginationItem, defaultProps);

exports.default = _default;
module.exports = exports.default;
