// @nextui-org/react/cjs/input/input.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _inputLabel = _interopRequireDefault(require("./input-label"));

var _inputBlockLabel = _interopRequireDefault(require("./input-block-label"));

var _inputContent = _interopRequireDefault(require("./input-content"));

var _inputIconClear = _interopRequireDefault(require("./input-icon-clear"));

var _styles = require("./styles");

var _collections = require("../utils/collections");

var _inputProps = require("./input-props");

var _dimensions = require("../utils/dimensions");

var _clsx = _interopRequireDefault(require("../utils/clsx"));

var _styles2 = require("../utils/styles");

var _assertion = require("../utils/assertion");

var _useWarning = _interopRequireDefault(require("../use-warning"));

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const simulateChangeEvent = (el, event) => {
  return { ...event,
    target: el,
    currentTarget: el
  };
};

const Input = /*#__PURE__*/_react.default.forwardRef(({
  as: Component = 'input',
  label,
  labelPlaceholder,
  labelLeft,
  labelRight,
  size,
  helperText,
  color: colorProp,
  helperColor: helperColorProp,
  status,
  contentLeft,
  contentRight,
  contentClickable,
  contentLeftStyling,
  contentRightStyling,
  onContentClick,
  initialValue,
  onChange,
  readOnly,
  value,
  onClearClick,
  clearable,
  shadow,
  animated,
  required,
  width: widthProp,
  className,
  onBlur,
  onFocus,
  autoComplete,
  placeholder,
  fullWidth,
  borderWeight: borderWeightProp,
  disabled,
  bordered,
  underlined,
  rounded,
  ...props
}, ref) => {
  const theme = (0, _useTheme.default)();
  const inputRef = (0, _react.useRef)(null);
  (0, _react.useImperativeHandle)(ref, () => inputRef.current);
  const [selfValue, setSelfValue] = (0, _react.useState)(initialValue);
  const [hover, setHover] = (0, _react.useState)(false);
  const {
    heightRatio,
    fontSize
  } = (0, _react.useMemo)(() => (0, _styles.getSizes)(size), [size]);
  const isControlledComponent = (0, _react.useMemo)(() => value !== undefined, [value]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = (0, _styles2.getFocusStyles)(theme);
  const inputLabel = (0, _react.useMemo)(() => label || labelPlaceholder, [label, labelPlaceholder]);
  const ComponentWrapper = (0, _react.useMemo)(() => inputLabel ? 'div' : 'label', [inputLabel]);
  const inputPlaceholder = (0, _react.useMemo)(() => labelPlaceholder ? '' : placeholder, [placeholder, labelPlaceholder]);
  const width = (0, _react.useMemo)(() => {
    if (fullWidth) return '100%';
    if (widthProp) return widthProp;
    return 'initial';
  }, [fullWidth, widthProp]);

  if (underlined && _assertion.__DEV__) {
    bordered && (0, _useWarning.default)('Using underlined and bordered at the same time will have no effect.');
    rounded && (0, _useWarning.default)('Using underlined and rounded at the same time will have no effect.');
  }

  const {
    bgColor,
    color,
    helperColor,
    placeholderColor,
    borderColor,
    hoverBorder,
    shadowColor
  } = (0, _react.useMemo)(() => (0, _styles.getColors)(theme, disabled, colorProp, status, helperColorProp), [theme.palette, theme.expressiveness, colorProp, helperColorProp, status, disabled]);
  const radius = (0, _react.useMemo)(() => (0, _dimensions.getNormalRadius)(size, rounded), [size, rounded]);
  const borderWeight = (0, _react.useMemo)(() => bordered || underlined ? (0, _dimensions.getNormalWeight)(borderWeightProp) : '0px', [bordered, underlined, borderWeightProp]);

  const changeHandler = event => {
    if (disabled || readOnly) return;
    setSelfValue(event.target.value);
    onChange && onChange(event);
  };

  const clearHandler = event => {
    setSelfValue('');
    onClearClick && onClearClick(event);
    /* istanbul ignore next */

    if (!inputRef.current) return;
    const changeEvent = simulateChangeEvent(inputRef.current, event);
    changeEvent.target.value = '';
    onChange && onChange(changeEvent);
    inputRef.current.focus();
  };

  const focusHandler = e => {
    setHover(true);
    onFocus && onFocus(e);
  };

  const blurHandler = e => {
    setHover(false);
    onBlur && onBlur(e);
  };

  const contentClickHandler = (key, e) => {
    if (disabled) return;
    onContentClick && onContentClick(key, e);
  };

  const contentProps = (0, _react.useMemo)(() => ({
    ratio: heightRatio,
    clickable: contentClickable,
    onClick: contentClickHandler
  }), [heightRatio, contentClickable, contentClickHandler]);
  (0, _react.useEffect)(() => {
    if (isControlledComponent) {
      setSelfValue(value);
    }
  });
  const isTextarea = (0, _react.useMemo)(() => Component === 'textarea', [Component]);
  const controlledValue = isControlledComponent ? {
    value: selfValue
  } : {
    defaultValue: initialValue
  };
  const inputProps = { ...props,
    ...controlledValue
  };
  const {
    inputId,
    labelId
  } = (0, _react.useMemo)(() => {
    const nextuiId = (0, _collections.getId)();
    return {
      inputId: inputProps.id || `next-ui-${nextuiId}`,
      labelId: !(0, _assertion.isEmpty)(inputProps.id) ? `${inputProps.id}-label` : `next-ui-${nextuiId}-label`
    };
  }, [inputProps.id]);

  if (inputLabel) {
    inputProps['aria-labelledby'] = labelId;
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    className: _style.default.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + "with-label",
    children: [inputLabel && /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputBlockLabel.default, {
      labelId: labelId,
      fontSize: fontSize,
      isTextarea: isTextarea,
      bordered: bordered,
      underlined: underlined,
      animated: animated,
      rounded: rounded,
      color: hoverBorder,
      status: status,
      hasLeftContent: !!contentLeft,
      selfValue: selfValue,
      heightRatio: heightRatio,
      asPlaceholder: !!labelPlaceholder,
      placeholderColor: placeholderColor,
      hover: hover,
      htmlFor: inputId,
      label: inputLabel
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      tabIndex: disabled ? -1 : 0,
      className: _style.default.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + ((0, _clsx.default)('container', {
        'input-container': !isTextarea,
        'textarea-container': isTextarea,
        'read-only': readOnly,
        hover
      }, className, focusClassName) || ""),
      children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(ComponentWrapper, {
        className: _style.default.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + ((0, _clsx.default)('wrapper', {
          hover,
          disabled,
          bordered,
          underlined,
          shadow,
          'input-wrapper': !isTextarea,
          'textarea-wrapper': isTextarea
        }) || ""),
        children: [labelLeft && /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputLabel.default, {
          status: status,
          bgColor: bgColor,
          borderWeight: borderWeight,
          underlined: underlined,
          bordered: bordered,
          color: placeholderColor,
          radius: radius,
          fontSize: fontSize,
          children: labelLeft
        }), contentLeft && /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputContent.default, {
          isLeft: true,
          applyStyles: contentLeftStyling,
          status: status,
          content: contentLeft,
          ...contentProps
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)(Component, {
          type: "text",
          id: inputId,
          ref: inputRef,
          placeholder: inputPlaceholder,
          disabled: disabled,
          readOnly: readOnly,
          onFocus: focusHandler,
          onBlur: blurHandler,
          onChange: changeHandler,
          autoComplete: autoComplete,
          "aria-placeholder": inputPlaceholder,
          "aria-readonly": readOnly,
          "aria-required": required,
          "aria-multiline": isTextarea,
          ...inputProps,
          className: _style.default.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + (inputProps && inputProps.className != null && inputProps.className || (0, _clsx.default)({
            disabled,
            rounded,
            'right-content': contentRight,
            'left-content': contentLeft
          }) || "")
        }), clearable && /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputIconClear.default, {
          status: status,
          underlined: underlined,
          visible: Boolean(selfValue),
          hasContentRight: !!contentRight,
          heightRatio: heightRatio,
          disabled: disabled || readOnly,
          onClick: clearHandler
        }), contentRight && /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputContent.default, {
          status: status,
          content: contentRight,
          applyStyles: contentRightStyling,
          ...contentProps
        }), labelRight && /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputLabel.default, {
          status: status,
          bgColor: bgColor,
          borderWeight: borderWeight,
          bordered: bordered,
          underlined: underlined,
          color: placeholderColor,
          radius: radius,
          fontSize: fontSize,
          isRight: true,
          children: labelRight
        })]
      })
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: _style.default.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + ((0, _clsx.default)('helper-text-container', {
        'with-value': !!helperText
      }) || ""),
      children: helperText && /*#__PURE__*/(0, _jsxRuntime.jsx)("p", {
        className: _style.default.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + "helper-text",
        children: helperText
      })
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "2127980039",
      dynamic: [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color],
      children: `.with-label.__jsx-style-dynamic-selector{width:${width};display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-box-align:center;border-radius:${radius};}.container.__jsx-style-dynamic-selector{width:100%;-webkit-transition:${animated ? 'all 0.25s ease' : 'none'};transition:${animated ? 'all 0.25s ease' : 'none'};border-radius:${radius};}.input-container.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:calc(${heightRatio} * ${theme.layout.gap});}.wrapper.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;vertical-align:middle;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;border-radius:${radius};background:${bgColor};}.input-wrapper.__jsx-style-dynamic-selector{height:100%;}.wrapper.shadow.__jsx-style-dynamic-selector{-webkit-transition:${animated ? 'all 0.25s ease' : 'none'};transition:${animated ? 'all 0.25s ease' : 'none'};}.wrapper.bordered.__jsx-style-dynamic-selector{background:transparent;border:none;box-shadow:0 0 0 ${!underlined ? borderWeight : '0px'} ${borderColor};-webkit-transition:${animated ? 'box-shadow 0.25s ease' : 'none'};transition:${animated ? 'box-shadow 0.25s ease' : 'none'};}.wrapper.underlined.__jsx-style-dynamic-selector{background:transparent;}.wrapper.underlined.__jsx-style-dynamic-selector::after{content:'';position:absolute;z-index:1;bottom:0;width:100%;height:${borderWeight};background:${borderColor};}.wrapper.underlined.__jsx-style-dynamic-selector::before{position:absolute;content:'';z-index:2;width:0;bottom:0;height:2px;left:50%;-webkit-transform:translate(-50%);-ms-transform:translate(-50%);transform:translate(-50%);background:${hoverBorder};-webkit-transition:${animated ? 'width 0.25s ease' : 'none'};transition:${animated ? 'width 0.25s ease' : 'none'};}.wrapper.hover.underlined.__jsx-style-dynamic-selector::before{width:100%;}.wrapper.disabled.__jsx-style-dynamic-selector{cursor:not-allowed;}.helper-text-container.__jsx-style-dynamic-selector{position:absolute;opacity:0;bottom:calc(${heightRatio} * ${theme.layout.gapHalf} * -1);-webkit-transition:${animated ? 'opacity 0.25s ease' : 'none'};transition:${animated ? 'opacity 0.25s ease' : 'none'};}.helper-text-container.with-value.__jsx-style-dynamic-selector{opacity:1;}.helper-text.__jsx-style-dynamic-selector{margin:2px 0 0 10px;font-size:0.7rem;color:${helperColor};}.container.hover.__jsx-style-dynamic-selector:not(.read-only){-webkit-transform:${animated && !underlined ? 'translateY(-2px)' : 'none'};-ms-transform:${animated && !underlined ? 'translateY(-2px)' : 'none'};transform:${animated && !underlined ? 'translateY(-2px)' : 'none'};}.wrapper.shadow.hover.__jsx-style-dynamic-selector:not(.read-only){box-shadow:${shadow && !underlined ? shadowColor : 'none'};}.container.__jsx-style-dynamic-selector:hover .wrapper.bordered.__jsx-style-dynamic-selector,.container.hover.__jsx-style-dynamic-selector:not(.read-only) .wrapper.bordered.__jsx-style-dynamic-selector{border-color:${hoverBorder};box-shadow:0 0 0 ${!underlined ? borderWeight : '0px'} ${hoverBorder};}input.disabled.__jsx-style-dynamic-selector,textarea.disabled.__jsx-style-dynamic-selector{color:${theme.palette.accents_4};cursor:not-allowed;}input.rounded.__jsx-style-dynamic-selector,textarea.rounded.__jsx-style-dynamic-selector{padding:0 ${theme.layout.gapQuarter};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-webkit-input-placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-webkit-input-placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-moz-placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-moz-placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector:-ms-input-placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector:-ms-input-placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}.wrapper.__jsx-style-dynamic-selector:not(.underlined) input.__jsx-style-dynamic-selector,.wrapper.__jsx-style-dynamic-selector:not(.underlined) textarea.__jsx-style-dynamic-selector{margin:4px 10px;}.wrapper.underlined.__jsx-style-dynamic-selector input.__jsx-style-dynamic-selector,.wrapper.underlined.__jsx-style-dynamic-selector textarea.__jsx-style-dynamic-selector{margin:4px 5px;}input.__jsx-style-dynamic-selector,textarea.__jsx-style-dynamic-selector{padding:0;font-size:${fontSize};background-color:transparent;border:none;color:${color};border-radius:0;width:100%;min-width:0;-webkit-appearance:none;}textarea.__jsx-style-dynamic-selector:not(.underlined){padding:${theme.layout.gapQuarter};}input.left-content.__jsx-style-dynamic-selector,textarea.left-content.__jsx-style-dynamic-selector{margin-left:0;}input.right-content.__jsx-style-dynamic-selector,textarea.right-content.__jsx-style-dynamic-selector{margin-right:0;}input.__jsx-style-dynamic-selector::-webkit-input-placeholder,textarea.__jsx-style-dynamic-selector::-webkit-input-placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector::-moz-placeholder,textarea.__jsx-style-dynamic-selector::-moz-placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:-ms-input-placeholder,textarea.__jsx-style-dynamic-selector:-ms-input-placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector::placeholder,textarea.__jsx-style-dynamic-selector::placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:-webkit-autofill,input.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:hover,input.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:active,input.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:focus,textarea.__jsx-style-dynamic-selector:-webkit-autofill,textarea.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:hover,textarea.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:active,textarea.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:focus{-webkit-box-shadow:0 0 0 30px ${theme.palette.background} inset !important;-webkit-text-fill-color:${color} !important;}`
    }), focusStyles]
  });
});

Input.defaultProps = _inputProps.defaultProps;
var _default = Input;
exports.default = _default;
module.exports = exports.default;
