// @nextui-org/react/cjs/input/input-label.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _clsx = _interopRequireDefault(require("../utils/clsx"));

var _color = require("../utils/color");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const InputLabel = ({
  children,
  isRight,
  fontSize,
  bgColor,
  radius,
  status,
  color,
  bordered,
  underlined,
  borderWeight,
  ...props
}) => {
  const theme = (0, _useTheme.default)();
  const isDark = theme.type === 'dark';
  const inputBgColor = (0, _react.useMemo)(() => {
    return underlined ? 'transparent' : status === 'default' && !bordered ? isDark ? theme.palette.accents_3 : (0, _color.addColorAlpha)(theme.palette.accents_3, 0.3) : bgColor || theme.palette.accents_2;
  }, [isDark, status, underlined, bgColor, bordered]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("span", { ...props,
    className: _style.default.dynamic([["4239241287", [inputBgColor, theme.layout.gapHalf, color || theme.palette.accents_4, fontSize, radius, radius, radius, radius, borderWeight, color || theme.palette.accents_4]]]) + " " + (props && props.className != null && props.className || (0, _clsx.default)('input-label', {
      right: isRight,
      left: !isRight,
      underlined,
      bordered
    }) || ""),
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "4239241287",
      dynamic: [inputBgColor, theme.layout.gapHalf, color || theme.palette.accents_4, fontSize, radius, radius, radius, radius, borderWeight, color || theme.palette.accents_4],
      children: `.input-label.__jsx-style-dynamic-selector{position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;width:initial;height:100%;font-weight:500;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;pointer-events:none;margin:0;background:${inputBgColor};padding:0 ${theme.layout.gapHalf};color:${color || theme.palette.accents_4};font-size:${fontSize};line-height:1;}.right.__jsx-style-dynamic-selector{border-top-right-radius:${radius};border-bottom-right-radius:${radius};}.left.__jsx-style-dynamic-selector{border-top-left-radius:${radius};border-bottom-left-radius:${radius};}.underlined.left.__jsx-style-dynamic-selector:after,.underlined.right.__jsx-style-dynamic-selector:before{position:absolute;content:'';height:70%;top:15%;bottom:0;width:calc(${borderWeight} / 2);box-shadow:0 2px 4px 0px rgb(0 0 0 / 8%);background:${color || theme.palette.accents_4};}.underlined.left.__jsx-style-dynamic-selector:after{right:0;}.underlined.right.__jsx-style-dynamic-selector:before{left:0;}.bordered.__jsx-style-dynamic-selector:after,.bordered.__jsx-style-dynamic-selector:before{display:none;}`
    })]
  });
};

const MemoInputLabel = /*#__PURE__*/_react.default.memo(InputLabel);

var _default = MemoInputLabel;
exports.default = _default;
module.exports = exports.default;
