// @nextui-org/react/esm/input/input.js
import _JSXStyle from "styled-jsx/style";
import React, { useEffect, useImperativeHandle, useMemo, useRef, useState } from 'react';
import useTheme from '../use-theme';
import InputLabel from './input-label';
import InputBlockLabel from './input-block-label';
import InputContent from './input-content';
import InputIconClear from './input-icon-clear';
import { getSizes, getColors } from './styles';
import { getId } from '../utils/collections';
import { defaultProps } from './input-props';
import { getNormalRadius, getNormalWeight } from '../utils/dimensions';
import clsx from '../utils/clsx';
import { getFocusStyles } from '../utils/styles';
import { isEmpty } from '../utils/assertion';
import useWarning from '../use-warning';
import { __DEV__ } from '../utils/assertion';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";

const simulateChangeEvent = (el, event) => {
  return { ...event,
    target: el,
    currentTarget: el
  };
};

const Input = /*#__PURE__*/React.forwardRef(({
  as: Component = 'input',
  label,
  labelPlaceholder,
  labelLeft,
  labelRight,
  size,
  helperText,
  color: colorProp,
  helperColor: helperColorProp,
  status,
  contentLeft,
  contentRight,
  contentClickable,
  contentLeftStyling,
  contentRightStyling,
  onContentClick,
  initialValue,
  onChange,
  readOnly,
  value,
  onClearClick,
  clearable,
  shadow,
  animated,
  required,
  width: widthProp,
  className,
  onBlur,
  onFocus,
  autoComplete,
  placeholder,
  fullWidth,
  borderWeight: borderWeightProp,
  disabled,
  bordered,
  underlined,
  rounded,
  ...props
}, ref) => {
  const theme = useTheme();
  const inputRef = useRef(null);
  useImperativeHandle(ref, () => inputRef.current);
  const [selfValue, setSelfValue] = useState(initialValue);
  const [hover, setHover] = useState(false);
  const {
    heightRatio,
    fontSize
  } = useMemo(() => getSizes(size), [size]);
  const isControlledComponent = useMemo(() => value !== undefined, [value]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = getFocusStyles(theme);
  const inputLabel = useMemo(() => label || labelPlaceholder, [label, labelPlaceholder]);
  const ComponentWrapper = useMemo(() => inputLabel ? 'div' : 'label', [inputLabel]);
  const inputPlaceholder = useMemo(() => labelPlaceholder ? '' : placeholder, [placeholder, labelPlaceholder]);
  const width = useMemo(() => {
    if (fullWidth) return '100%';
    if (widthProp) return widthProp;
    return 'initial';
  }, [fullWidth, widthProp]);

  if (underlined && __DEV__) {
    bordered && useWarning('Using underlined and bordered at the same time will have no effect.');
    rounded && useWarning('Using underlined and rounded at the same time will have no effect.');
  }

  const {
    bgColor,
    color,
    helperColor,
    placeholderColor,
    borderColor,
    hoverBorder,
    shadowColor
  } = useMemo(() => getColors(theme, disabled, colorProp, status, helperColorProp), [theme.palette, theme.expressiveness, colorProp, helperColorProp, status, disabled]);
  const radius = useMemo(() => getNormalRadius(size, rounded), [size, rounded]);
  const borderWeight = useMemo(() => bordered || underlined ? getNormalWeight(borderWeightProp) : '0px', [bordered, underlined, borderWeightProp]);

  const changeHandler = event => {
    if (disabled || readOnly) return;
    setSelfValue(event.target.value);
    onChange && onChange(event);
  };

  const clearHandler = event => {
    setSelfValue('');
    onClearClick && onClearClick(event);
    /* istanbul ignore next */

    if (!inputRef.current) return;
    const changeEvent = simulateChangeEvent(inputRef.current, event);
    changeEvent.target.value = '';
    onChange && onChange(changeEvent);
    inputRef.current.focus();
  };

  const focusHandler = e => {
    setHover(true);
    onFocus && onFocus(e);
  };

  const blurHandler = e => {
    setHover(false);
    onBlur && onBlur(e);
  };

  const contentClickHandler = (key, e) => {
    if (disabled) return;
    onContentClick && onContentClick(key, e);
  };

  const contentProps = useMemo(() => ({
    ratio: heightRatio,
    clickable: contentClickable,
    onClick: contentClickHandler
  }), [heightRatio, contentClickable, contentClickHandler]);
  useEffect(() => {
    if (isControlledComponent) {
      setSelfValue(value);
    }
  });
  const isTextarea = useMemo(() => Component === 'textarea', [Component]);
  const controlledValue = isControlledComponent ? {
    value: selfValue
  } : {
    defaultValue: initialValue
  };
  const inputProps = { ...props,
    ...controlledValue
  };
  const {
    inputId,
    labelId
  } = useMemo(() => {
    const nextuiId = getId();
    return {
      inputId: inputProps.id || `next-ui-${nextuiId}`,
      labelId: !isEmpty(inputProps.id) ? `${inputProps.id}-label` : `next-ui-${nextuiId}-label`
    };
  }, [inputProps.id]);

  if (inputLabel) {
    inputProps['aria-labelledby'] = labelId;
  }

  return /*#__PURE__*/_jsxs("div", {
    className: _JSXStyle.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + "with-label",
    children: [inputLabel && /*#__PURE__*/_jsx(InputBlockLabel, {
      labelId: labelId,
      fontSize: fontSize,
      isTextarea: isTextarea,
      bordered: bordered,
      underlined: underlined,
      animated: animated,
      rounded: rounded,
      color: hoverBorder,
      status: status,
      hasLeftContent: !!contentLeft,
      selfValue: selfValue,
      heightRatio: heightRatio,
      asPlaceholder: !!labelPlaceholder,
      placeholderColor: placeholderColor,
      hover: hover,
      htmlFor: inputId,
      label: inputLabel
    }), /*#__PURE__*/_jsx("div", {
      tabIndex: disabled ? -1 : 0,
      className: _JSXStyle.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + (clsx('container', {
        'input-container': !isTextarea,
        'textarea-container': isTextarea,
        'read-only': readOnly,
        hover
      }, className, focusClassName) || ""),
      children: /*#__PURE__*/_jsxs(ComponentWrapper, {
        className: _JSXStyle.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + (clsx('wrapper', {
          hover,
          disabled,
          bordered,
          underlined,
          shadow,
          'input-wrapper': !isTextarea,
          'textarea-wrapper': isTextarea
        }) || ""),
        children: [labelLeft && /*#__PURE__*/_jsx(InputLabel, {
          status: status,
          bgColor: bgColor,
          borderWeight: borderWeight,
          underlined: underlined,
          bordered: bordered,
          color: placeholderColor,
          radius: radius,
          fontSize: fontSize,
          children: labelLeft
        }), contentLeft && /*#__PURE__*/_jsx(InputContent, {
          isLeft: true,
          applyStyles: contentLeftStyling,
          status: status,
          content: contentLeft,
          ...contentProps
        }), /*#__PURE__*/_jsx(Component, {
          type: "text",
          id: inputId,
          ref: inputRef,
          placeholder: inputPlaceholder,
          disabled: disabled,
          readOnly: readOnly,
          onFocus: focusHandler,
          onBlur: blurHandler,
          onChange: changeHandler,
          autoComplete: autoComplete,
          "aria-placeholder": inputPlaceholder,
          "aria-readonly": readOnly,
          "aria-required": required,
          "aria-multiline": isTextarea,
          ...inputProps,
          className: _JSXStyle.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + (inputProps && inputProps.className != null && inputProps.className || clsx({
            disabled,
            rounded,
            'right-content': contentRight,
            'left-content': contentLeft
          }) || "")
        }), clearable && /*#__PURE__*/_jsx(InputIconClear, {
          status: status,
          underlined: underlined,
          visible: Boolean(selfValue),
          hasContentRight: !!contentRight,
          heightRatio: heightRatio,
          disabled: disabled || readOnly,
          onClick: clearHandler
        }), contentRight && /*#__PURE__*/_jsx(InputContent, {
          status: status,
          content: contentRight,
          applyStyles: contentRightStyling,
          ...contentProps
        }), labelRight && /*#__PURE__*/_jsx(InputLabel, {
          status: status,
          bgColor: bgColor,
          borderWeight: borderWeight,
          bordered: bordered,
          underlined: underlined,
          color: placeholderColor,
          radius: radius,
          fontSize: fontSize,
          isRight: true,
          children: labelRight
        })]
      })
    }), /*#__PURE__*/_jsx("div", {
      className: _JSXStyle.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + (clsx('helper-text-container', {
        'with-value': !!helperText
      }) || ""),
      children: helperText && /*#__PURE__*/_jsx("p", {
        className: _JSXStyle.dynamic([["2127980039", [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color]]]) + " " + "helper-text",
        children: helperText
      })
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "2127980039",
      dynamic: [width, radius, animated ? 'all 0.25s ease' : 'none', radius, heightRatio, theme.layout.gap, radius, bgColor, animated ? 'all 0.25s ease' : 'none', !underlined ? borderWeight : '0px', borderColor, animated ? 'box-shadow 0.25s ease' : 'none', borderWeight, borderColor, hoverBorder, animated ? 'width 0.25s ease' : 'none', heightRatio, theme.layout.gapHalf, animated ? 'opacity 0.25s ease' : 'none', helperColor, animated && !underlined ? 'translateY(-2px)' : 'none', shadow && !underlined ? shadowColor : 'none', hoverBorder, !underlined ? borderWeight : '0px', hoverBorder, theme.palette.accents_4, theme.layout.gapQuarter, animated ? 'opacity 0.25s ease 0s' : 'none', fontSize, color, theme.layout.gapQuarter, placeholderColor, animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', animated ? 'opacity 0.25s ease 0s' : 'none', theme.palette.background, color],
      children: `.with-label.__jsx-style-dynamic-selector{width:${width};display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-box-align:center;border-radius:${radius};}.container.__jsx-style-dynamic-selector{width:100%;-webkit-transition:${animated ? 'all 0.25s ease' : 'none'};transition:${animated ? 'all 0.25s ease' : 'none'};border-radius:${radius};}.input-container.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:calc(${heightRatio} * ${theme.layout.gap});}.wrapper.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;vertical-align:middle;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;border-radius:${radius};background:${bgColor};}.input-wrapper.__jsx-style-dynamic-selector{height:100%;}.wrapper.shadow.__jsx-style-dynamic-selector{-webkit-transition:${animated ? 'all 0.25s ease' : 'none'};transition:${animated ? 'all 0.25s ease' : 'none'};}.wrapper.bordered.__jsx-style-dynamic-selector{background:transparent;border:none;box-shadow:0 0 0 ${!underlined ? borderWeight : '0px'} ${borderColor};-webkit-transition:${animated ? 'box-shadow 0.25s ease' : 'none'};transition:${animated ? 'box-shadow 0.25s ease' : 'none'};}.wrapper.underlined.__jsx-style-dynamic-selector{background:transparent;}.wrapper.underlined.__jsx-style-dynamic-selector::after{content:'';position:absolute;z-index:1;bottom:0;width:100%;height:${borderWeight};background:${borderColor};}.wrapper.underlined.__jsx-style-dynamic-selector::before{position:absolute;content:'';z-index:2;width:0;bottom:0;height:2px;left:50%;-webkit-transform:translate(-50%);-ms-transform:translate(-50%);transform:translate(-50%);background:${hoverBorder};-webkit-transition:${animated ? 'width 0.25s ease' : 'none'};transition:${animated ? 'width 0.25s ease' : 'none'};}.wrapper.hover.underlined.__jsx-style-dynamic-selector::before{width:100%;}.wrapper.disabled.__jsx-style-dynamic-selector{cursor:not-allowed;}.helper-text-container.__jsx-style-dynamic-selector{position:absolute;opacity:0;bottom:calc(${heightRatio} * ${theme.layout.gapHalf} * -1);-webkit-transition:${animated ? 'opacity 0.25s ease' : 'none'};transition:${animated ? 'opacity 0.25s ease' : 'none'};}.helper-text-container.with-value.__jsx-style-dynamic-selector{opacity:1;}.helper-text.__jsx-style-dynamic-selector{margin:2px 0 0 10px;font-size:0.7rem;color:${helperColor};}.container.hover.__jsx-style-dynamic-selector:not(.read-only){-webkit-transform:${animated && !underlined ? 'translateY(-2px)' : 'none'};-ms-transform:${animated && !underlined ? 'translateY(-2px)' : 'none'};transform:${animated && !underlined ? 'translateY(-2px)' : 'none'};}.wrapper.shadow.hover.__jsx-style-dynamic-selector:not(.read-only){box-shadow:${shadow && !underlined ? shadowColor : 'none'};}.container.__jsx-style-dynamic-selector:hover .wrapper.bordered.__jsx-style-dynamic-selector,.container.hover.__jsx-style-dynamic-selector:not(.read-only) .wrapper.bordered.__jsx-style-dynamic-selector{border-color:${hoverBorder};box-shadow:0 0 0 ${!underlined ? borderWeight : '0px'} ${hoverBorder};}input.disabled.__jsx-style-dynamic-selector,textarea.disabled.__jsx-style-dynamic-selector{color:${theme.palette.accents_4};cursor:not-allowed;}input.rounded.__jsx-style-dynamic-selector,textarea.rounded.__jsx-style-dynamic-selector{padding:0 ${theme.layout.gapQuarter};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-webkit-input-placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-webkit-input-placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-moz-placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::-moz-placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector:-ms-input-placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector:-ms-input-placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::placeholder,textarea.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector::placeholder{opacity:0;-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}.wrapper.__jsx-style-dynamic-selector:not(.underlined) input.__jsx-style-dynamic-selector,.wrapper.__jsx-style-dynamic-selector:not(.underlined) textarea.__jsx-style-dynamic-selector{margin:4px 10px;}.wrapper.underlined.__jsx-style-dynamic-selector input.__jsx-style-dynamic-selector,.wrapper.underlined.__jsx-style-dynamic-selector textarea.__jsx-style-dynamic-selector{margin:4px 5px;}input.__jsx-style-dynamic-selector,textarea.__jsx-style-dynamic-selector{padding:0;font-size:${fontSize};background-color:transparent;border:none;color:${color};border-radius:0;width:100%;min-width:0;-webkit-appearance:none;}textarea.__jsx-style-dynamic-selector:not(.underlined){padding:${theme.layout.gapQuarter};}input.left-content.__jsx-style-dynamic-selector,textarea.left-content.__jsx-style-dynamic-selector{margin-left:0;}input.right-content.__jsx-style-dynamic-selector,textarea.right-content.__jsx-style-dynamic-selector{margin-right:0;}input.__jsx-style-dynamic-selector::-webkit-input-placeholder,textarea.__jsx-style-dynamic-selector::-webkit-input-placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector::-moz-placeholder,textarea.__jsx-style-dynamic-selector::-moz-placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:-ms-input-placeholder,textarea.__jsx-style-dynamic-selector:-ms-input-placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector::placeholder,textarea.__jsx-style-dynamic-selector::placeholder{color:${placeholderColor};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-moz-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-ms-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};-webkit-transition:${animated ? 'opacity 0.25s ease 0s' : 'none'};}input.__jsx-style-dynamic-selector:-webkit-autofill,input.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:hover,input.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:active,input.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:focus,textarea.__jsx-style-dynamic-selector:-webkit-autofill,textarea.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:hover,textarea.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:active,textarea.__jsx-style-dynamic-selector:-webkit-autofill.__jsx-style-dynamic-selector:focus{-webkit-box-shadow:0 0 0 30px ${theme.palette.background} inset !important;-webkit-text-fill-color:${color} !important;}`
    }), focusStyles]
  });
});
Input.defaultProps = defaultProps;
export default Input;
