// @nextui-org/react/cjs/link/link.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireWildcard(require("react"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _icon = _interopRequireDefault(require("./icon"));

var _color = require("../utils/color");

var _assertion = require("../utils/assertion");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  href: '',
  icon: false,
  underline: false,
  block: false,
  className: ''
};

const Link = /*#__PURE__*/_react.default.forwardRef(({
  href,
  color,
  underline,
  children,
  className,
  block,
  icon,
  ...props
}, ref) => {
  const theme = (0, _useTheme.default)();
  const linkColor = (0, _react.useMemo)(() => (0, _color.getNormalColor)(color || block, theme.palette, theme.palette.link), [color, theme.palette]);
  const hoverColor = (0, _react.useMemo)(() => color || block ? (0, _color.addColorAlpha)(linkColor, 0.8) : (0, _color.addColorAlpha)(theme.palette.text, 0.8), [color, block, theme, linkColor]);
  const padding = block ? theme.layout.gapQuarter : '0';
  const decoration = underline ? 'underline' : 'none';
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("a", {
    href: href,
    ...props,
    ref: ref,
    className: _style.default.dynamic([["250630921", [linkColor, padding, padding, block ? theme.layout.radius : 0, decoration, block ? (0, _color.addColorAlpha)(linkColor, 0.2) : 'inherit', hoverColor]]]) + " " + (props && props.className != null && props.className || `link ${className}`),
    children: [children, icon && /*#__PURE__*/(0, _jsxRuntime.jsx)(_icon.default, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "250630921",
      dynamic: [linkColor, padding, padding, block ? theme.layout.radius : 0, decoration, block ? (0, _color.addColorAlpha)(linkColor, 0.2) : 'inherit', hoverColor],
      children: `.link.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:baseline;-webkit-box-align:baseline;-ms-flex-align:baseline;align-items:baseline;line-height:inherit;color:${linkColor};-webkit-text-decoration:none;text-decoration:none;padding:calc(${padding} * 0.8) calc(${padding} * 1.7);border-radius:${block ? theme.layout.radius : 0};width:-webkit-fit-content;width:-moz-fit-content;width:fit-content;-webkit-transition:all 0.25s ease;transition:all 0.25s ease;}.link.__jsx-style-dynamic-selector:hover,.link.__jsx-style-dynamic-selector:active,.link.__jsx-style-dynamic-selector:focus{-webkit-text-decoration:${decoration};text-decoration:${decoration};}.link.__jsx-style-dynamic-selector:hover{background-color:${block ? (0, _color.addColorAlpha)(linkColor, 0.2) : 'inherit'};color:${hoverColor};}`
    })]
  });
});

if (_assertion.__DEV__) {
  Link.displayName = 'Link';
}

const MemoLink = /*#__PURE__*/_react.default.memo(Link);

var _default = (0, _withDefaults.default)(MemoLink, defaultProps);

exports.default = _default;
module.exports = exports.default;
