// @nextui-org/react/esm/collapse/collapse.js
import _JSXStyle from "styled-jsx/style";
import React, { useEffect, useMemo } from 'react';
import CollapseIcon from './collapse-icon';
import useTheme from '../use-theme';
import Expand from '../utils/expand';
import { useCollapseContext } from './collapse-context';
import useCurrentState from '../use-current-state';
import useWarning from '../use-warning';
import { getNormalWeight } from '../utils/dimensions';
import { getId } from '../utils/collections';
import { getFocusStyles } from '../utils/styles';
import useKeyboard, { KeyCode } from '../use-keyboard';
import clsx from '../utils/clsx';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const defaultProps = {
  className: '',
  shadow: false,
  divider: true,
  bordered: false,
  showArrow: true,
  animated: true,
  disabled: false,
  borderWeight: 'light',
  initialExpanded: false
};

const Collapse = ({
  children,
  title,
  subtitle,
  initialExpanded,
  shadow,
  className,
  divider,
  arrowIcon,
  showArrow,
  disabled,
  onChange,
  bordered,
  contentLeft,
  animated: animatedProp,
  borderWeight: borderWeightProp,
  index,
  ...props
}) => {
  const theme = useTheme();
  const [visible, setVisible, visibleRef] = useCurrentState(initialExpanded);
  const {
    values,
    divider: groupDivider,
    animated: groupAnimated,
    updateValues
  } = useCollapseContext();

  if (!title) {
    useWarning('"title" is required.', 'Collapse');
  }

  useEffect(() => {
    if (!values.length) return;
    const isActive = !!values.find(item => item === index);
    setVisible(isActive);
  }, [values.join(',')]);
  const {
    className: focusClassName,
    styles: focusStyles
  } = getFocusStyles(theme);
  const arrowComponent = useMemo(() => {
    if (!showArrow) return null;
    return arrowIcon ? arrowIcon : /*#__PURE__*/_jsx(CollapseIcon, {});
  }, [arrowIcon, showArrow]);
  const borderWeight = useMemo(() => {
    const withDivider = groupDivider === undefined ? divider : groupDivider;
    return withDivider ? getNormalWeight(borderWeightProp) : '0px';
  }, [divider, groupDivider, borderWeightProp]);
  const animated = useMemo(() => {
    return groupAnimated === undefined ? animatedProp : groupAnimated;
  }, [groupAnimated, animatedProp]);
  const bgColor = useMemo(() => theme.type === 'dark' ? theme.palette.accents_1 : theme.palette.background, [theme.type]);
  const {
    ariaLabelledById,
    ariaControlId
  } = useMemo(() => {
    const nextuiId = getId();
    return {
      ariaLabelledById: `collapse-button-next-ui-${nextuiId}`,
      ariaControlId: `collapse-next-ui-${nextuiId}`
    };
  }, []);

  const handleChange = event => {
    if (disabled) return;
    const next = !visibleRef.current;
    setVisible(next);
    updateValues && updateValues(index, next);
    onChange && onChange(event, index, next);
  };

  const {
    bindings
  } = useKeyboard(event => {
    handleChange(event);
  }, [KeyCode.Enter, KeyCode.Space], {
    disableGlobalEvent: true
  });
  return /*#__PURE__*/_jsxs("div", {
    tabIndex: disabled ? -1 : 0,
    ...props,
    ...bindings,
    className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + (bindings && bindings.className != null && bindings.className || props && props.className != null && props.className || clsx('collapse', {
      shadow,
      bordered
    }, className, focusClassName) || ""),
    children: [/*#__PURE__*/_jsx("div", {
      role: "button",
      id: ariaLabelledById,
      "aria-disabled": disabled,
      "aria-expanded": visible,
      "aria-controls": ariaControlId,
      onClick: handleChange,
      className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + (clsx('view', {
        disabled
      }) || ""),
      children: /*#__PURE__*/_jsxs("div", {
        className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + (clsx('title', {
          animated
        }) || ""),
        children: [contentLeft && /*#__PURE__*/_jsx("div", {
          className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + "title-content-left",
          children: contentLeft
        }), /*#__PURE__*/_jsxs("div", {
          className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + "title-content",
          children: [/*#__PURE__*/React.isValidElement(title) ? title : /*#__PURE__*/_jsx("h3", {
            className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]),
            children: title
          }), subtitle && /*#__PURE__*/_jsx("div", {
            className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + "subtitle",
            children: subtitle
          })]
        }), /*#__PURE__*/_jsx("div", {
          className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + "title-content-right",
          children: arrowComponent
        })]
      })
    }), /*#__PURE__*/_jsx(Expand, {
      isExpanded: visible,
      animated: animated,
      children: /*#__PURE__*/_jsx("div", {
        role: "region",
        tabIndex: -1,
        id: ariaControlId,
        "aria-labelledby": ariaLabelledById,
        className: _JSXStyle.dynamic([["4287550624", [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap]]]) + " " + "content",
        children: children
      })
    }), /*#__PURE__*/_jsx(_JSXStyle, {
      id: "4287550624",
      dynamic: [borderWeight, theme.palette.border, borderWeight, theme.palette.border, bgColor, theme.expressiveness.shadowMedium, theme.layout.radius, theme.layout.gap, theme.layout.gap, theme.layout.radius, borderWeight, theme.palette.border, theme.layout.gap, theme.layout.gapHalf, theme.palette.foreground, visible ? '-90deg' : '0', theme.palette.accents_5, theme.layout.gap],
      children: `.collapse.__jsx-style-dynamic-selector{border-top:${borderWeight} solid ${theme.palette.border};border-bottom:${borderWeight} solid ${theme.palette.border};-webkit-transition:box-shadow 0.25s ease;transition:box-shadow 0.25s ease;}.shadow.__jsx-style-dynamic-selector{border:none;background:${bgColor};box-shadow:${theme.expressiveness.shadowMedium};border-radius:${theme.layout.radius};padding:0 ${theme.layout.gap};}.bordered.__jsx-style-dynamic-selector{padding:0 ${theme.layout.gap};border-radius:${theme.layout.radius};border:${borderWeight} solid ${theme.palette.border};}.view.__jsx-style-dynamic-selector{width:100%;display:block;text-align:left;background:transparent;border:none;cursor:pointer;padding:${theme.layout.gap} 0;}.view.disabled.__jsx-style-dynamic-selector{cursor:not-allowed;}.view.disabled.__jsx-style-dynamic-selector .title.__jsx-style-dynamic-selector,.view.disabled.__jsx-style-dynamic-selector .subtitle.__jsx-style-dynamic-selector{opacity:0.5;}.title-content-left.__jsx-style-dynamic-selector,.title-content-right.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.title-content-left.__jsx-style-dynamic-selector{margin-right:${theme.layout.gapHalf};}.title-content.__jsx-style-dynamic-selector{width:100%;}.title.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:${theme.palette.foreground};}.title-content-right.__jsx-style-dynamic-selector svg{-webkit-transform:rotateZ(${visible ? '-90deg' : '0'});-ms-transform:rotateZ(${visible ? '-90deg' : '0'});transform:rotateZ(${visible ? '-90deg' : '0'});}.animated.__jsx-style-dynamic-selector .title-content-right.__jsx-style-dynamic-selector svg{-webkit-transition:-webkit-transform 200ms ease;-webkit-transition:transform 200ms ease;transition:transform 200ms ease;}.title-content.__jsx-style-dynamic-selector h1,.title-content.__jsx-style-dynamic-selector h2,.title-content.__jsx-style-dynamic-selector h3,.title-content.__jsx-style-dynamic-selector h4,.title-content.__jsx-style-dynamic-selector h5,.title-content.__jsx-style-dynamic-selector h6,.title-content.__jsx-style-dynamic-selector p,.title-content.__jsx-style-dynamic-selector span,.title-content.__jsx-style-dynamic-selector b{margin:0 !important;}.subtitle.__jsx-style-dynamic-selector{color:${theme.palette.accents_5};margin:0;}.subtitle.__jsx-style-dynamic-selector>*{margin:0;}.content.__jsx-style-dynamic-selector{font-size:1rem;line-height:1.625rem;padding-bottom:${theme.layout.gap};}.content.__jsx-style-dynamic-selector>*:first-child{margin-top:0;}.content.__jsx-style-dynamic-selector>*:last-child{margin-bottom:0;}`
    }), focusStyles]
  });
};

Collapse.defaultProps = defaultProps;
export default Collapse;
