// @nextui-org/react/cjs/row/row.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _react = _interopRequireDefault(require("react"));

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _useTheme = _interopRequireDefault(require("../use-theme"));

var _jsxRuntime = require("react/jsx-runtime");

const defaultProps = {
  gap: 0,
  fluid: true,
  wrap: 'nowrap',
  justify: 'flex-start',
  align: 'flex-start',
  as: 'div',
  className: ''
};

const Row = ({
  children,
  as,
  gap,
  fluid,
  justify,
  align,
  wrap,
  className,
  ...props
}) => {
  const Component = as;
  const theme = (0, _useTheme.default)();
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(Component, { ...props,
    className: _style.default.dynamic([["575978181", [wrap, gap, theme.layout.gap, gap, theme.layout.gap, gap, theme.layout.gap, justify, align]]]) + " " + (props && props.className != null && props.className || `row ${className} ${fluid && 'fluid'}`),
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_style.default, {
      id: "575978181",
      dynamic: [wrap, gap, theme.layout.gap, gap, theme.layout.gap, gap, theme.layout.gap, justify, align],
      children: `.row.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;position:relative;box-sizing:border-box;-webkit-flex-wrap:${wrap};-ms-flex-wrap:${wrap};flex-wrap:${wrap};margin-left:calc(${gap} * ${theme.layout.gap} / 2);margin-right:calc(${gap} * ${theme.layout.gap} / 2);--row-gap:calc(${gap} * ${theme.layout.gap});-webkit-box-pack:${justify};-webkit-justify-content:${justify};-ms-flex-pack:${justify};justify-content:${justify};-webkit-align-items:${align};-webkit-box-align:${align};-ms-flex-align:${align};align-items:${align};}.fluid.__jsx-style-dynamic-selector{width:100%;}`
    })]
  });
};

const MemoRow = /*#__PURE__*/_react.default.memo(Row);

var _default = (0, _withDefaults.default)(MemoRow, defaultProps);

exports.default = _default;
module.exports = exports.default;
